<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>3&nbsp; Data containers – Orchestrating Microbiome Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../pages/import.html" rel="next">
<link href="../pages/miaverse.html" rel="prev">
<link href="../assets/favicon.png" rel="icon" type="image/png">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
</head>
<body class="nav-sidebar docked nav-fixed slimcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../assets/mia_logo.png" alt="" class="navbar-logo"></a>
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Orchestrating Microbiome Analysis</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-bi-github">
<li>
    <a class="dropdown-item" href="https://github.com/microbiome/OMA/">
 <span class="dropdown-text">Source Code</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://microbiome.github.io/OMA/docs/devel/">
 <span class="dropdown-text">Browse version <code>devel</code></span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com"> <i class="bi bi-twitter" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
</div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../pages/containers.html">Data containers &amp; importing</a></li><li class="breadcrumb-item"><a href="../pages/containers.html"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Data containers</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto"><div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Overview</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false">
 <span class="menu-text">Introduction</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Microbiome data science in Bioconductor</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/miaverse.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">miaverse</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Data containers &amp; importing</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/containers.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Data containers</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/import.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Import</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/convert.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Convert &amp; export</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false">
 <span class="menu-text">Data wrangling</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/taxonomy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Taxonomic information</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/wrangling.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Data wrangling</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/subsetting.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Subsetting</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/agglomeration.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Agglomeration</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/transformation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Transformation</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false">
 <span class="menu-text">Exploration &amp; QC</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/quality_control.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Exploration &amp; quality control</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/composition.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Community composition</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false">
 <span class="menu-text">Diversity &amp; similarity</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/alpha_diversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Community diversity</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/beta_diversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Community similarity</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/clustering.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">Community typing</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" role="navigation" aria-expanded="false">
 <span class="menu-text">Association</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/differential_abundance.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">16</span>&nbsp; <span class="chapter-title">Differential abundance</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/correlation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">17</span>&nbsp; <span class="chapter-title">Correlation</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" role="navigation" aria-expanded="false">
 <span class="menu-text">Networks</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-7" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/network_learning.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">18</span>&nbsp; <span class="chapter-title">Network learning &amp; analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/network_comparison.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">19</span>&nbsp; <span class="chapter-title">Network comparison</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" role="navigation" aria-expanded="false">
 <span class="menu-text">Multiomics</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-8" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/cross_correlation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">20</span>&nbsp; <span class="chapter-title">Cross-association</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/multiassay_ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">21</span>&nbsp; <span class="chapter-title">Ordination-based multiassay analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/integrated_learner.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">22</span>&nbsp; <span class="chapter-title">Multi-omics prediction and classification</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-9" role="navigation" aria-expanded="false">
 <span class="menu-text">Machine learning &amp; statistical modeling</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-9" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-9" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/machine_learning.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">23</span>&nbsp; <span class="chapter-title">Machine learning</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/statistical_modeling.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">24</span>&nbsp; <span class="chapter-title">Statistical modeling</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-10" role="navigation" aria-expanded="false">
 <span class="menu-text">Workflows</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-10" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-10" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/introductory_workflow.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">25</span>&nbsp; <span class="chapter-title">Introductory</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/introductory_workflow_french_version.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">26</span>&nbsp; <span class="chapter-title">Introductory (French)</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/introductory_workflow_dutch_version.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">27</span>&nbsp; <span class="chapter-title">Introductory (Dutch)</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-11" role="navigation" aria-expanded="false">
 <span class="menu-text">Training</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-11" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-11" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/training.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">28</span>&nbsp; <span class="chapter-title">Training</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/exercises.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">29</span>&nbsp; <span class="chapter-title">Exercises</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-12" role="navigation" aria-expanded="false">
 <span class="menu-text">Support &amp; resources</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-12" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-12" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/support.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">30</span>&nbsp; <span class="chapter-title">Support</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/resources.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">31</span>&nbsp; <span class="chapter-title">Resources</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/acknowledgments.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">32</span>&nbsp; <span class="chapter-title">Contributions</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-13" role="navigation" aria-expanded="false">
 <span class="menu-text">Appendices</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-13" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-13" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/msea.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">A</span>&nbsp; <span class="chapter-title">Microbe Set Enrichment Analysis (MSEA)</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/mmuphin_meta_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">B</span>&nbsp; <span class="chapter-title">Meta-analyses</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/extra_material.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">C</span>&nbsp; <span class="chapter-title">Extra material</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/visualization.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">D</span>&nbsp; <span class="chapter-title">Visualization</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/session_info.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">E</span>&nbsp; <span class="chapter-title">Technical info</span></span></a>
  </div>
</li>
      </ul>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">Table of contents</h2>
   
  <ul>
<li><a href="#sec-rows-and-cols" id="toc-sec-rows-and-cols" class="nav-link active" data-scroll-target="#sec-rows-and-cols"><span class="header-section-number">3.1</span> Rows and columns</a></li>
  <li><a href="#sec-assay-slot" id="toc-sec-assay-slot" class="nav-link" data-scroll-target="#sec-assay-slot"><span class="header-section-number">3.2</span> Assay data</a></li>
  <li><a href="#coldata" id="toc-coldata" class="nav-link" data-scroll-target="#coldata"><span class="header-section-number">3.3</span> colData</a></li>
  <li><a href="#sec-rowData" id="toc-sec-rowData" class="nav-link" data-scroll-target="#sec-rowData"><span class="header-section-number">3.4</span> rowData</a></li>
  <li><a href="#rowtree" id="toc-rowtree" class="nav-link" data-scroll-target="#rowtree"><span class="header-section-number">3.5</span> rowTree</a></li>
  <li><a href="#sec-alt-exp" id="toc-sec-alt-exp" class="nav-link" data-scroll-target="#sec-alt-exp"><span class="header-section-number">3.6</span> Alternative Experiments</a></li>
  <li><a href="#sec-mae" id="toc-sec-mae" class="nav-link" data-scroll-target="#sec-mae"><span class="header-section-number">3.7</span> Multiple experiments</a></li>
  </ul></nav>
    </div>
<!-- main -->
<main class="content page-columns page-full" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../pages/containers.html">Data containers &amp; importing</a></li><li class="breadcrumb-item"><a href="../pages/containers.html"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Data containers</span></a></li></ol></nav><div class="quarto-title">
<h1 class="title"><span id="sec-containers" class="quarto-section-identifier"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Data containers</span></span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header><div class="cell" data-layout-align="center">
<script>
document.addEventListener("click", function (event) {
    if (event.target.classList.contains("rebook-collapse")) {
        event.target.classList.toggle("active");
        var content = event.target.nextElementSibling;
        if (content.style.display === "block") {
            content.style.display = "none";
        } else {
            content.style.display = "block";
        }
    }
})
</script><style>
.rebook-collapse {
  background-color: #eee;
  color: #444;
  cursor: pointer;
  padding: 18px;
  width: 100%;
  border: none;
  text-align: left;
  outline: none;
  font-size: 15px;
}

.rebook-content {
  padding: 0 18px;
  display: none;
  overflow: hidden;
  background-color: #f1f1f1;
}
</style>
</div>
<p>This section provides an introduction to <code>TreeSummarizedExperiment (TreeSE)</code> and <code>MultiAssayExperiment (MAE)</code> data containers introduced in <a href="intro.html#sec-microbiome-bioc" class="quarto-xref"><span>Section 1.2</span></a>. In microbiome data science, these containers link taxonomic abundance tables with rich side information on the features and samples. Taxonomic abundance data can be obtained by 16S rRNA amplicon or metagenomic sequencing, phylogenetic microarrays, or by other means. Many microbiome experiments include multiple versions and types of data generated independently or derived from each other through transformation or agglomeration. We start by providing recommendations on how to represent different varieties of multi-table data within the <code>TreeSE</code> class.</p>
<p>The options and recommendations are summarized in <a href="#tbl-options" class="quarto-xref">Table&nbsp;<span>3.1</span></a>.</p>
<section id="sec-rows-and-cols" class="level2" data-number="3.1"><h2 data-number="3.1" class="anchored" data-anchor-id="sec-rows-and-cols">
<span class="header-section-number">3.1</span> Rows and columns</h2>
<p>Let us load example data and store it in variable <code>tse</code>.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/microbiome/mia">mia</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"GlobalPatterns"</span>, package <span class="op">=</span> <span class="st">"mia"</span><span class="op">)</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="va">GlobalPatterns</span></span>
<span><span class="va">tse</span></span>
<span><span class="co">##  class: TreeSummarizedExperiment </span></span>
<span><span class="co">##  dim: 19216 26 </span></span>
<span><span class="co">##  metadata(0):</span></span>
<span><span class="co">##  assays(1): counts</span></span>
<span><span class="co">##  rownames(19216): 549322 522457 ... 200359 271582</span></span>
<span><span class="co">##  rowData names(7): Kingdom Phylum ... Genus Species</span></span>
<span><span class="co">##  colnames(26): CL3 CC1 ... Even2 Even3</span></span>
<span><span class="co">##  colData names(7): X.SampleID Primer ... SampleType Description</span></span>
<span><span class="co">##  reducedDimNames(0):</span></span>
<span><span class="co">##  mainExpName: NULL</span></span>
<span><span class="co">##  altExpNames(0):</span></span>
<span><span class="co">##  rowLinks: a LinkDataFrame (19216 rows)</span></span>
<span><span class="co">##  rowTree: 1 phylo tree(s) (19216 leaves)</span></span>
<span><span class="co">##  colLinks: NULL</span></span>
<span><span class="co">##  colTree: NULL</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The <code>TreeSE</code> object, similar to a standard <code>data.frame</code> or <code>matrix</code>, has rows and columns. Typically, samples are stored in columns, while features or taxa are stored in rows. You can extract subsets of the data, such as the first five rows and certain three columns. The object manages the linkages between data, ensuring, for example, that when you subset the data, for instance, both the assay and sample metadata are subsetted simultaneously, ensuring they remain matched with each other.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="va">tse</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">19</span>, <span class="fl">16</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">tse</span></span>
<span><span class="co">##  class: TreeSummarizedExperiment </span></span>
<span><span class="co">##  dim: 5 3 </span></span>
<span><span class="co">##  metadata(0):</span></span>
<span><span class="co">##  assays(1): counts</span></span>
<span><span class="co">##  rownames(5): 549322 522457 951 244423 586076</span></span>
<span><span class="co">##  rowData names(7): Kingdom Phylum ... Genus Species</span></span>
<span><span class="co">##  colnames(3): CL3 TRRsed1 NP2</span></span>
<span><span class="co">##  colData names(7): X.SampleID Primer ... SampleType Description</span></span>
<span><span class="co">##  reducedDimNames(0):</span></span>
<span><span class="co">##  mainExpName: NULL</span></span>
<span><span class="co">##  altExpNames(0):</span></span>
<span><span class="co">##  rowLinks: a LinkDataFrame (5 rows)</span></span>
<span><span class="co">##  rowTree: 1 phylo tree(s) (19216 leaves)</span></span>
<span><span class="co">##  colLinks: NULL</span></span>
<span><span class="co">##  colTree: NULL</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Compared to the original data the dimensions are for rows and columns 5 and 3, respectively.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p><code>SummarizedExperiment</code> objects have rows and columns. Also <code>MultiAssayExperiment</code>, introduced in <a href="#sec-mae" class="quarto-xref"><span>Section 3.7</span></a> have rows and cols but the structure is more complicated. You can find more examples on subsetting from <a href="subsetting.html" class="quarto-xref"><span>Chapter 8</span></a>.</p>
</div>
</div>
</section><section id="sec-assay-slot" class="level2" data-number="3.2"><h2 data-number="3.2" class="anchored" data-anchor-id="sec-assay-slot">
<span class="header-section-number">3.2</span> Assay data</h2>
<p>The microbiome is the collection of all microbes (such as bacteria, viruses, fungi, etc.) in the body. When studying these microbes, abundance data is needed, and that’s where assays come in.</p>
<p>An assay is a way of measuring the presence and abundance of different types of microbes in a sample. For example, if you want to know how many bacteria of a certain type are in your gut, you can use an assay to measure this. When storing assays, the original data is count-based. However, the original count-based taxonomic abundance tables may undergo different transformations, such as logarithmic, Centered Log-Ratio (CLR), or relative abundance. These are typically stored in <em><strong>assays</strong></em>. See <a href="transformation.html" class="quarto-xref"><span>Chapter 10</span></a> for more information on transformations.</p>
<p>The <code>assays</code> slot contains the experimental data as multiple count matrices. The result of <code>assays</code> is a list of matrices.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assays</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span>
<span><span class="co">##  List of length 1</span></span>
<span><span class="co">##  names(1): counts</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Individual assays can be accessed via <code>assay</code>.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">tse</span>, <span class="st">"counts"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">##         CL3 TRRsed1 NP2</span></span>
<span><span class="co">##  549322   0       0   1</span></span>
<span><span class="co">##  522457   0       0   0</span></span>
<span><span class="co">##  951      0       0   0</span></span>
<span><span class="co">##  244423   0       0   0</span></span>
<span><span class="co">##  586076   0       0   0</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>So, in summary, in the world of microbiome analysis, an assay is essentially a way to quantify and understand the composition of microbes in a given sample, which is super important for all kinds of research, ranging from human health to environment studies.</p>
<p>Furthermore, to illustrate the use of multiple assays, we can create an empty matrix and add it to the object.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span><span class="op">(</span>nrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span>, ncol <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">tse</span>, <span class="st">"empty_table"</span>, withDimnames<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">mat</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assays</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span>
<span><span class="co">##  List of length 2</span></span>
<span><span class="co">##  names(2): counts empty_table</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now there are two assays available in the <code>tse</code> object, <code>counts</code> and <code>empty_table</code>.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">tse</span>, <span class="st">"empty_table"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">##         CL3 TRRsed1 NP2</span></span>
<span><span class="co">##  549322  NA      NA  NA</span></span>
<span><span class="co">##  522457  NA      NA  NA</span></span>
<span><span class="co">##  951     NA      NA  NA</span></span>
<span><span class="co">##  244423  NA      NA  NA</span></span>
<span><span class="co">##  586076  NA      NA  NA</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Here the dimension of the assay data remains unchanged. This is in fact a requirement for the assays.</p>
</section><section id="coldata" class="level2" data-number="3.3"><h2 data-number="3.3" class="anchored" data-anchor-id="coldata">
<span class="header-section-number">3.3</span> colData</h2>
<p><code>colData</code> contains information about the samples used in the study. This information can include details such as the sample ID, the primers used in the analysis, the barcodes associated with the sample (truncated or complete), the type of sample (e.g.&nbsp;soil, fecal, mock) and a description of the sample.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span>
<span><span class="co">##  DataFrame with 3 rows and 7 columns</span></span>
<span><span class="co">##          X.SampleID   Primer Final_Barcode Barcode_truncated_plus_T</span></span>
<span><span class="co">##            &lt;factor&gt; &lt;factor&gt;      &lt;factor&gt;                 &lt;factor&gt;</span></span>
<span><span class="co">##  CL3        CL3      ILBC_01        AACGCA                   TGCGTT</span></span>
<span><span class="co">##  TRRsed1    TRRsed1  ILBC_22        ACATGT                   ACATGT</span></span>
<span><span class="co">##  NP2        NP2      ILBC_19        ACAGTT                   AACTGT</span></span>
<span><span class="co">##          Barcode_full_length         SampleType</span></span>
<span><span class="co">##                     &lt;factor&gt;           &lt;factor&gt;</span></span>
<span><span class="co">##  CL3             CTAGCGTGCGT Soil              </span></span>
<span><span class="co">##  TRRsed1         CACGTGACATG Sediment (estuary)</span></span>
<span><span class="co">##  NP2             TCGCGCAACTG Ocean             </span></span>
<span><span class="co">##                                       Description</span></span>
<span><span class="co">##                                          &lt;factor&gt;</span></span>
<span><span class="co">##  CL3     Calhoun South Carolina Pine soil, pH 4.9</span></span>
<span><span class="co">##  TRRsed1 Tijuana River Reserve, depth 1          </span></span>
<span><span class="co">##  NP2     Newport Pier, CA surface water, Time 1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>To illustrate, <strong><em>X.SampleID</em></strong> gives the sample identifier, <strong><em>SampleType</em></strong> indicates the sample type (e.g.&nbsp;soil, fecal matter, control) and <strong><em>Description</em></strong> provides an additional description of the sample.</p>
</section><section id="sec-rowData" class="level2" data-number="3.4"><h2 data-number="3.4" class="anchored" data-anchor-id="sec-rowData">
<span class="header-section-number">3.4</span> rowData</h2>
<p><code>rowData</code> contains data on the features of the analyzed samples. This is particularly important in the microbiome field for storing taxonomic information. This taxonomic information is extremely important for understanding the composition and diversity of the microbiome in each sample analyzed. It enables identification of the different types of microorganisms present in samples. It also allows you to explore the relationships between microbiome composition and various environmental or health factors.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span>
<span><span class="co">##  DataFrame with 5 rows and 7 columns</span></span>
<span><span class="co">##             Kingdom        Phylum        Class        Order        Family</span></span>
<span><span class="co">##         &lt;character&gt;   &lt;character&gt;  &lt;character&gt;  &lt;character&gt;   &lt;character&gt;</span></span>
<span><span class="co">##  549322     Archaea Crenarchaeota Thermoprotei           NA            NA</span></span>
<span><span class="co">##  522457     Archaea Crenarchaeota Thermoprotei           NA            NA</span></span>
<span><span class="co">##  951        Archaea Crenarchaeota Thermoprotei Sulfolobales Sulfolobaceae</span></span>
<span><span class="co">##  244423     Archaea Crenarchaeota        Sd-NA           NA            NA</span></span>
<span><span class="co">##  586076     Archaea Crenarchaeota        Sd-NA           NA            NA</span></span>
<span><span class="co">##               Genus                Species</span></span>
<span><span class="co">##         &lt;character&gt;            &lt;character&gt;</span></span>
<span><span class="co">##  549322          NA                     NA</span></span>
<span><span class="co">##  522457          NA                     NA</span></span>
<span><span class="co">##  951     Sulfolobus Sulfolobusacidocalda..</span></span>
<span><span class="co">##  244423          NA                     NA</span></span>
<span><span class="co">##  586076          NA                     NA</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="rowtree" class="level2" data-number="3.5"><h2 data-number="3.5" class="anchored" data-anchor-id="rowtree">
<span class="header-section-number">3.5</span> rowTree</h2>
<p>Phylogenetic trees also play an important role in the microbiome field. The <code>TreeSE</code> class can keep track of features and node relations via two functions, <code>rowTree</code> and <code>rowLinks</code>.</p>
<p>A tree can be accessed via <code>rowTree</code> as <code>phylo</code> object.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/TreeSummarizedExperiment-accessor.html">rowTree</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##  Phylogenetic tree with 19216 tips and 19215 internal nodes.</span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##  Tip labels:</span></span>
<span><span class="co">##    549322, 522457, 951, 244423, 586076, 246140, ...</span></span>
<span><span class="co">##  Node labels:</span></span>
<span><span class="co">##    , 0.858.4, 1.000.154, 0.764.3, 0.995.2, 1.000.2, ...</span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##  Rooted; includes branch lengths.</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The links to the individual features are available through <code>rowLinks</code>.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/TreeSummarizedExperiment-accessor.html">rowLinks</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span>
<span><span class="co">##  LinkDataFrame with 5 rows and 5 columns</span></span>
<span><span class="co">##        nodeLab   nodeNum nodeLab_alias    isLeaf   whichTree</span></span>
<span><span class="co">##    &lt;character&gt; &lt;integer&gt;   &lt;character&gt; &lt;logical&gt; &lt;character&gt;</span></span>
<span><span class="co">##  1      549322         1       alias_1      TRUE       phylo</span></span>
<span><span class="co">##  2      522457         2       alias_2      TRUE       phylo</span></span>
<span><span class="co">##  3         951         3       alias_3      TRUE       phylo</span></span>
<span><span class="co">##  4      244423         4       alias_4      TRUE       phylo</span></span>
<span><span class="co">##  5      586076         5       alias_5      TRUE       phylo</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Please note that there can be a 1:1 relationship between tree nodes and features, but this is not a must-have. This means there can be features that are not linked to nodes, and nodes that are not linked to features. To change the links in an existing object, the <code><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/changeTree.html">changeTree()</a></code> function is available.</p>
</section><section id="sec-alt-exp" class="level2 page-columns page-full" data-number="3.6"><h2 data-number="3.6" class="anchored" data-anchor-id="sec-alt-exp">
<span class="header-section-number">3.6</span> Alternative Experiments</h2>
<p><em><strong>Alternative experiments</strong></em> complement <em>assays</em>. They can contain complementary data, which is no longer tied to the same dimensions as the assay data. However, the number of samples (columns) must be the same.</p>
<p>This can come into play, for instance, when one has taxonomic abundance profiles quantified using different measurement technologies, such as phylogenetic microarrays, amplicon sequencing, or metagenomic sequencing. Another common use case is including abundance tables for different taxonomic ranks. Such alternative experiments concerning the same set of samples can be stored as</p>
<ol type="1">
<li>Separate <em>assays</em> assuming that the taxonomic information can be mapped between features directly 1:1; or</li>
<li>Data in the <code>altExp</code> slot of the <code>TreeSE</code>, if the feature dimensions differ. Each element of the <code>altExp</code> slot is a <code>SE</code> or an object from a derived class with independent feature data.</li>
</ol>
<p>The following shows how to store taxonomic abundance tables agglomerated at different taxonomic levels. However, the data could as well originate from entirely different measurement sources as long as the samples match.</p>
<p>Let us first subset the data so that it has only two rows.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tse_sub</span> <span class="op">&lt;-</span> <span class="va">tse</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, <span class="op">]</span></span>
<span><span class="co"># Both have the same number of columns (samples)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span>
<span><span class="co">##  [1] 5 3</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">tse_sub</span><span class="op">)</span></span>
<span><span class="co">##  [1] 2 3</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Then we can add the new data object as an alternative experiment in the original data.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Add the new data object to the original data object as an alternative</span></span>
<span><span class="co"># experiment with the name "Phylum"</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html">altExp</a></span><span class="op">(</span><span class="va">tse</span>, <span class="st">"subsetted"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">tse_sub</span></span>
<span></span>
<span><span class="co"># Check the alternative experiment names available in the data</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html">altExpNames</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span>
<span><span class="co">##  [1] "subsetted"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We can now subset the data by taking certain samples, for instance, and this acts on both <code>altExp</code> and assay data.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tse_single_sample</span> <span class="op">&lt;-</span> <span class="va">tse</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html">altExp</a></span><span class="op">(</span><span class="va">tse_single_sample</span>,<span class="st">"subsetted"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">##  [1] 2 1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>For more details on <em>altExp</em>, you can check the <a href="https://bioconductor.org/packages/release/bioc/vignettes/SingleCellExperiment/inst/doc/intro.html">introduction</a> to the <code>SingleCellExperiment</code> package <span class="citation" data-cites="R_SingleCellExperiment">(<a href="#ref-R_SingleCellExperiment" role="doc-biblioref">Lun and Risso 2020</a>)</span>.</p>
<div class="no-row-height column-margin column-container"><div id="ref-R_SingleCellExperiment" class="csl-entry" role="listitem">
Lun, Aaron, and Davide Risso. 2020. <em>SingleCellExperiment: S4 Classes for Single Cell Data</em>.
</div></div></section><section id="sec-mae" class="level2 page-columns page-full" data-number="3.7"><h2 data-number="3.7" class="anchored" data-anchor-id="sec-mae">
<span class="header-section-number">3.7</span> Multiple experiments</h2>
<p><em><strong>Multiple experiments</strong></em> relate to complementary measurement types, such as transcriptomic or metabolomic profiling of the microbiome or the host. Multiple experiments can be represented using the same options as alternative experiments, or by using the <code>MAE</code> class <span class="citation" data-cites="Ramos2017">(<a href="#ref-Ramos2017" role="doc-biblioref">Ramos et al. 2017</a>)</span>. Depending on how the datasets relate to each other the data can be stored as:</p>
<div class="no-row-height column-margin column-container"><div id="ref-Ramos2017" class="csl-entry" role="listitem">
Ramos, Marcel, Lucas Schiffer, Angela Re, Rimsha Azhar, Azfar Basunia, Carmen Rodriguez Cabrera, Tiffany Chan, et al. 2017. <span>“Software for the Integration of Multiomics Experiments in Bioconductor.”</span> <em>Cancer Research</em>. <a href="https://doi.org/10.1158/0008-5472.CAN-17-0344">https://doi.org/10.1158/0008-5472.CAN-17-0344</a>.
</div></div><ol type="1">
<li>
<code>altExp</code> if the samples can be matched directly 1:1; or</li>
<li>As <code>MAE</code> objects, in which the connections between samples are defined through a <code>sampleMap</code>. Each element on the <code>ExperimentList</code> of an <code>MAE</code> is <code>matrix</code> or <code>matrix</code>-like objects, including <code>SE</code> objects, and the number of samples can differ between the elements.</li>
</ol>
<p>The <code>MAE</code> object can handle more complex relationships between experiments. It manages the linkages between samples and experiments, ensuring that the data remains consistent and well-organized.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"HintikkaXOData"</span><span class="op">)</span></span>
<span><span class="va">mae</span> <span class="op">&lt;-</span> <span class="va">HintikkaXOData</span></span>
<span><span class="va">mae</span></span>
<span><span class="co">##  A MultiAssayExperiment object of 3 listed</span></span>
<span><span class="co">##   experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##   Containing an ExperimentList class object of length 3:</span></span>
<span><span class="co">##   [1] microbiota: TreeSummarizedExperiment with 12706 rows and 40 columns</span></span>
<span><span class="co">##   [2] metabolites: TreeSummarizedExperiment with 38 rows and 40 columns</span></span>
<span><span class="co">##   [3] biomarkers: TreeSummarizedExperiment with 39 rows and 40 columns</span></span>
<span><span class="co">##  Functionality:</span></span>
<span><span class="co">##   experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##   colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##   sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##   `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##   *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##   assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##   exportClass() - save data to flat files</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The <code>sampleMap</code> is a crucial component of the <code>MAE</code> object as it acts as the important bookkeeper, maintaining the information about which samples are associated with which experiments. This ensures that data linkages are correctly managed and preserved across different types of experiments.</p>
<p>In fact, we can have</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://github.com/waldronlab/MultiAssayExperiment/reference/MultiAssayExperiment-methods.html">sampleMap</a></span><span class="op">(</span><span class="va">mae</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">##  DataFrame with 6 rows and 3 columns</span></span>
<span><span class="co">##         assay     primary     colname</span></span>
<span><span class="co">##      &lt;factor&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">##  1 microbiota          C1          C1</span></span>
<span><span class="co">##  2 microbiota          C2          C2</span></span>
<span><span class="co">##  3 microbiota          C3          C3</span></span>
<span><span class="co">##  4 microbiota          C4          C4</span></span>
<span><span class="co">##  5 microbiota          C5          C5</span></span>
<span><span class="co">##  6 microbiota          C6          C6</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>For illustration, let’s subset the data by taking first five samples.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mae</span> <span class="op">&lt;-</span> <span class="va">mae</span><span class="op">[</span> , <span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="op">]</span></span>
<span><span class="va">mae</span></span>
<span><span class="co">##  A MultiAssayExperiment object of 3 listed</span></span>
<span><span class="co">##   experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##   Containing an ExperimentList class object of length 3:</span></span>
<span><span class="co">##   [1] microbiota: TreeSummarizedExperiment with 12706 rows and 5 columns</span></span>
<span><span class="co">##   [2] metabolites: TreeSummarizedExperiment with 38 rows and 5 columns</span></span>
<span><span class="co">##   [3] biomarkers: TreeSummarizedExperiment with 39 rows and 5 columns</span></span>
<span><span class="co">##  Functionality:</span></span>
<span><span class="co">##   experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##   colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##   sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##   `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##   *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##   assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##   exportClass() - save data to flat files</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>If you have multiple experiments containing multiple measures from same patients, you can utilize the <code>MultiAssayExperiment</code> object to keep track of which samples belong to which patient.</p>
</div>
</div>
<p>The following dataset illustrates how to utilize the sample mapping system in <code>MAE</code>. It includes two omics: biogenic amines and fatty acids, collected from 10 chickens.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mae</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"mae_holofood.Rds"</span>, package <span class="op">=</span> <span class="st">"OMA"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mae</span></span>
<span><span class="co">##  A MultiAssayExperiment object of 2 listed</span></span>
<span><span class="co">##   experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##   Containing an ExperimentList class object of length 2:</span></span>
<span><span class="co">##   [1] BIOGENIC AMINES: TreeSummarizedExperiment with 7 rows and 14 columns</span></span>
<span><span class="co">##   [2] FATTY ACIDS: TreeSummarizedExperiment with 19 rows and 19 columns</span></span>
<span><span class="co">##  Functionality:</span></span>
<span><span class="co">##   experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##   colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##   sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##   `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##   *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##   assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##   exportClass() - save data to flat files</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We can see that there are more than ten samples per omic dataset due to multiple time points collected for some animals. From the <code>colData</code> of <code>MAE</code>, we can observe the animal metadata shared between omics and time points, including information that remains constant throughout the trial.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">mae</span><span class="op">)</span></span>
<span><span class="co">##  DataFrame with 10 rows and 13 columns</span></span>
<span><span class="co">##                 marker.canonical_url         marker.type Treatment code</span></span>
<span><span class="co">##                            &lt;numeric&gt;              &lt;list&gt;    &lt;character&gt;</span></span>
<span><span class="co">##  SAMEA112904734                   NA TREATMENT,TRIAL,PEN             CO</span></span>
<span><span class="co">##  SAMEA112904735                   NA TREATMENT,TRIAL,PEN             CC</span></span>
<span><span class="co">##  SAMEA112904737                   NA TREATMENT,TRIAL,PEN             CC</span></span>
<span><span class="co">##  SAMEA112904741                   NA TREATMENT,TRIAL,PEN             CC</span></span>
<span><span class="co">##  SAMEA112904746                   NA TREATMENT,TRIAL,PEN             CC</span></span>
<span><span class="co">##  SAMEA112904748                   NA TREATMENT,TRIAL,PEN             CO</span></span>
<span><span class="co">##  SAMEA112904749                   NA TREATMENT,TRIAL,PEN             CC</span></span>
<span><span class="co">##  SAMEA112904755                   NA TREATMENT,TRIAL,PEN             CC</span></span>
<span><span class="co">##  SAMEA112904762                   NA TREATMENT,TRIAL,PEN             CO</span></span>
<span><span class="co">##  SAMEA112904763                   NA TREATMENT,TRIAL,PEN             CC</span></span>
<span><span class="co">##                 Treatment name  Trial code Trial description   Trial end</span></span>
<span><span class="co">##                    &lt;character&gt; &lt;character&gt;       &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">##  SAMEA112904734      Probiotic          CB           Trial 2  2019-05-19</span></span>
<span><span class="co">##  SAMEA112904735        Control          CC           Trial 3  2019-07-14</span></span>
<span><span class="co">##  SAMEA112904737        Control          CB           Trial 2  2019-05-19</span></span>
<span><span class="co">##  SAMEA112904741        Control          CC           Trial 3  2019-07-14</span></span>
<span><span class="co">##  SAMEA112904746        Control          CB           Trial 2  2019-05-19</span></span>
<span><span class="co">##  SAMEA112904748      Probiotic          CC           Trial 3  2019-07-14</span></span>
<span><span class="co">##  SAMEA112904749        Control          CA           Trial 1  2019-03-11</span></span>
<span><span class="co">##  SAMEA112904755        Control          CB           Trial 2  2019-05-19</span></span>
<span><span class="co">##  SAMEA112904762      Probiotic          CC           Trial 3  2019-07-14</span></span>
<span><span class="co">##  SAMEA112904763        Control          CA           Trial 1  2019-03-11</span></span>
<span><span class="co">##                 Trial start         animal      system          canonical_url</span></span>
<span><span class="co">##                 &lt;character&gt;    &lt;character&gt; &lt;character&gt;            &lt;character&gt;</span></span>
<span><span class="co">##  SAMEA112904734  2019-04-21 SAMEA112904734     chicken https://www.ebi.ac.u..</span></span>
<span><span class="co">##  SAMEA112904735  2019-06-09 SAMEA112904735     chicken https://www.ebi.ac.u..</span></span>
<span><span class="co">##  SAMEA112904737  2019-04-21 SAMEA112904737     chicken https://www.ebi.ac.u..</span></span>
<span><span class="co">##  SAMEA112904741  2019-06-09 SAMEA112904741     chicken https://www.ebi.ac.u..</span></span>
<span><span class="co">##  SAMEA112904746  2019-04-21 SAMEA112904746     chicken https://www.ebi.ac.u..</span></span>
<span><span class="co">##  SAMEA112904748  2019-06-09 SAMEA112904748     chicken https://www.ebi.ac.u..</span></span>
<span><span class="co">##  SAMEA112904749  2019-02-04 SAMEA112904749     chicken https://www.ebi.ac.u..</span></span>
<span><span class="co">##  SAMEA112904755  2019-04-21 SAMEA112904755     chicken https://www.ebi.ac.u..</span></span>
<span><span class="co">##  SAMEA112904762  2019-06-09 SAMEA112904762     chicken https://www.ebi.ac.u..</span></span>
<span><span class="co">##  SAMEA112904763  2019-02-04 SAMEA112904763     chicken https://www.ebi.ac.u..</span></span>
<span><span class="co">##                 Average Daily Feed intake: Day 00 - 07</span></span>
<span><span class="co">##                                              &lt;numeric&gt;</span></span>
<span><span class="co">##  SAMEA112904734                                  19.39</span></span>
<span><span class="co">##  SAMEA112904735                                  22.35</span></span>
<span><span class="co">##  SAMEA112904737                                  18.18</span></span>
<span><span class="co">##  SAMEA112904741                                  21.68</span></span>
<span><span class="co">##  SAMEA112904746                                  19.75</span></span>
<span><span class="co">##  SAMEA112904748                                  21.10</span></span>
<span><span class="co">##  SAMEA112904749                                  18.67</span></span>
<span><span class="co">##  SAMEA112904755                                  23.36</span></span>
<span><span class="co">##  SAMEA112904762                                  21.15</span></span>
<span><span class="co">##  SAMEA112904763                                  19.67</span></span>
<span><span class="co">##                 Average Daily Feed intake: Day 00 - 07, unit</span></span>
<span><span class="co">##                                                  &lt;character&gt;</span></span>
<span><span class="co">##  SAMEA112904734                                            g</span></span>
<span><span class="co">##  SAMEA112904735                                            g</span></span>
<span><span class="co">##  SAMEA112904737                                            g</span></span>
<span><span class="co">##  SAMEA112904741                                            g</span></span>
<span><span class="co">##  SAMEA112904746                                            g</span></span>
<span><span class="co">##  SAMEA112904748                                            g</span></span>
<span><span class="co">##  SAMEA112904749                                            g</span></span>
<span><span class="co">##  SAMEA112904755                                            g</span></span>
<span><span class="co">##  SAMEA112904762                                            g</span></span>
<span><span class="co">##  SAMEA112904763                                            g</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The <code>sampleMap</code> slot now contains mappings between each sample and the corresponding animal. There are as many rows as there are total samples.</p>
<p>The “colname” column refers to the samples in the omic dataset identified in the “assay” column, while the “primary” column provides information about the animals. You will notice that some animals are listed multiple times, reflecting the multiple omics and time points collected for those individuals.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://github.com/waldronlab/MultiAssayExperiment/reference/MultiAssayExperiment-methods.html">sampleMap</a></span><span class="op">(</span><span class="va">mae</span><span class="op">)</span></span>
<span><span class="co">##  DataFrame with 33 rows and 3 columns</span></span>
<span><span class="co">##                assay        primary        colname</span></span>
<span><span class="co">##             &lt;factor&gt;    &lt;character&gt;    &lt;character&gt;</span></span>
<span><span class="co">##  1   BIOGENIC AMINES SAMEA112904734 SAMEA112906114</span></span>
<span><span class="co">##  2   BIOGENIC AMINES SAMEA112904734 SAMEA112906592</span></span>
<span><span class="co">##  3   BIOGENIC AMINES SAMEA112904735 SAMEA112906338</span></span>
<span><span class="co">##  4   BIOGENIC AMINES SAMEA112904735 SAMEA112906810</span></span>
<span><span class="co">##  5   BIOGENIC AMINES SAMEA112904737 SAMEA112906123</span></span>
<span><span class="co">##  ...             ...            ...            ...</span></span>
<span><span class="co">##  29      FATTY ACIDS SAMEA112904755 SAMEA112906112</span></span>
<span><span class="co">##  30      FATTY ACIDS SAMEA112904755 SAMEA112906590</span></span>
<span><span class="co">##  31      FATTY ACIDS SAMEA112904762 SAMEA112906315</span></span>
<span><span class="co">##  32      FATTY ACIDS SAMEA112904762 SAMEA112906787</span></span>
<span><span class="co">##  33      FATTY ACIDS SAMEA112904763 SAMEA112905916</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>For information have a look at the <a href="https://bioconductor.org/packages/release/bioc/vignettes/MultiAssayExperiment/inst/doc/MultiAssayExperiment.html">intro vignette</a> of the <code>MultiAssayExperiment</code> package.</p>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Recommended options for storing multiple data tables in microbiome studies
</div>
</div>
<div class="callout-body-container callout-body">
<div id="tbl-options" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure"><figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-options-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;3.1: The <em>assays</em> are best suited for data transformations (one-to-one match between samples and columns across the assays). The <em>alternative experiments</em> are particularly suitable for alternative versions of the data that are of same type but may have a different number of features (e.g.&nbsp;taxonomic groups); this is for instance the case with taxonomic abundance tables agglomerated at different levels (e.g.&nbsp;genus vs.&nbsp;phyla) or alternative profiling technologies (e.g.&nbsp;amplicon sequencing vs.&nbsp;shallow shotgun metagenomics). For alternative experiments one-to-one match between samples (cols) is libraryd but the alternative experiment tables can have different numbers of features (rows). Finally, elements of the <em>MAE</em> provide the most flexible way to incorporate multi-omic data tables with flexible numbers of samples and features. We recommend these conventions as the basis for methods development and application in microbiome studies.
</figcaption><div aria-describedby="tbl-options-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<table class="caption-top table">
<thead><tr class="header">
<th style="text-align: right;">Option</th>
<th>Rows (features)</th>
<th style="text-align: right;">Cols (samples)</th>
<th style="text-align: right;">Recommended</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">assays</td>
<td>match</td>
<td style="text-align: right;">match</td>
<td style="text-align: right;">Data transformations</td>
</tr>
<tr class="even">
<td style="text-align: right;">altExp</td>
<td>free</td>
<td style="text-align: right;">match</td>
<td style="text-align: right;">Alternative experiments</td>
</tr>
<tr class="odd">
<td style="text-align: right;">MultiAssay</td>
<td>free</td>
<td style="text-align: right;">free (mapping)</td>
<td style="text-align: right;">Multi-omic experiments</td>
</tr>
</tbody>
</table>
</div>
</figure>
</div>
<p>Multi-assay analyses, discussed in sections <a href="cross_correlation.html#sec-cross-correlation" class="quarto-xref"><span>Section 20.1</span></a> and <a href="multiassay_ordination.html" class="quarto-xref"><span>Chapter 21</span></a>, can be facilitated by the multi-assay data containers, <em>TreeSummarizedExperiment</em> and <em>MultiAssayExperiment</em>. These are scalable and contain different types of data in a single container, making this framework particularly suited for multi-assay microbiome data incorporating different types of complementary data sources in a single, reproducible workflow. An alternative experiment can be stored in <em>altExp</em> slot of the SE data container. Alternatively, both experiments can be stored side-by-side in an <code>MAE</code> data container.</p>
</div>
</div>



</section><a onclick="window.scrollTo(0, 0); return false;" role="button" id="quarto-back-to-top"><i class="bi bi-arrow-up"></i> Back to top</a></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = false;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="../pages/miaverse.html" class="pagination-link" aria-label="miaverse">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">miaverse</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../pages/import.html" class="pagination-link" aria-label="Import">
        <span class="nav-page-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Import</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
<p><a href="./">Orchestrating Microbiome Analysis with Bioconductor</a></p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">
<p>This book was built with <a href="https://github.com/js2264/BiocBook/">BiocBook</a> with <span class="emoji" data-emoji="heart">❤️</span></p>
</div>
  </div>
</footer>


</body></html>