<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>4&nbsp; Import – Orchestrating Microbiome Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../pages/convert.html" rel="next">
<link href="../pages/containers.html" rel="prev">
<link href="../assets/favicon.png" rel="icon" type="image/png">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
</head>
<body class="nav-sidebar docked nav-fixed slimcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../assets/mia_logo.png" alt="" class="navbar-logo"></a>
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Orchestrating Microbiome Analysis</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-bi-github">
<li>
    <a class="dropdown-item" href="https://github.com/microbiome/OMA/">
 <span class="dropdown-text">Source Code</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://microbiome.github.io/OMA/docs/devel/">
 <span class="dropdown-text">Browse version <code>devel</code></span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com"> <i class="bi bi-twitter" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
</div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../pages/containers.html">Data containers &amp; importing</a></li><li class="breadcrumb-item"><a href="../pages/import.html"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Import</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto"><div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Overview</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false">
 <span class="menu-text">Introduction</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Microbiome data science in Bioconductor</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/miaverse.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">miaverse</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Data containers &amp; importing</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/containers.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Data containers</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/import.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Import</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/convert.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Convert &amp; export</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false">
 <span class="menu-text">Data wrangling</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/taxonomy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Taxonomic information</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/wrangling.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Data wrangling</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/subsetting.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Subsetting</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/agglomeration.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Agglomeration</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/transformation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Transformation</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false">
 <span class="menu-text">Exploration &amp; QC</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/quality_control.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Exploration &amp; quality control</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/composition.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Community composition</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false">
 <span class="menu-text">Diversity &amp; similarity</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/alpha_diversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Community diversity</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/beta_diversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Community similarity</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/clustering.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">Community typing</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" role="navigation" aria-expanded="false">
 <span class="menu-text">Association</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/differential_abundance.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">16</span>&nbsp; <span class="chapter-title">Differential abundance</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/correlation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">17</span>&nbsp; <span class="chapter-title">Correlation</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" role="navigation" aria-expanded="false">
 <span class="menu-text">Networks</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-7" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/network_learning.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">18</span>&nbsp; <span class="chapter-title">Network learning &amp; analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/network_comparison.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">19</span>&nbsp; <span class="chapter-title">Network comparison</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" role="navigation" aria-expanded="false">
 <span class="menu-text">Multiomics</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-8" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/cross_correlation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">20</span>&nbsp; <span class="chapter-title">Cross-association</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/multiassay_ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">21</span>&nbsp; <span class="chapter-title">Ordination-based multiassay analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/integrated_learner.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">22</span>&nbsp; <span class="chapter-title">Multi-omics prediction and classification</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-9" role="navigation" aria-expanded="false">
 <span class="menu-text">Machine learning &amp; statistical modeling</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-9" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-9" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/machine_learning.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">23</span>&nbsp; <span class="chapter-title">Machine learning</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/statistical_modeling.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">24</span>&nbsp; <span class="chapter-title">Statistical modeling</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-10" role="navigation" aria-expanded="false">
 <span class="menu-text">Workflows</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-10" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-10" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/introductory_workflow.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">25</span>&nbsp; <span class="chapter-title">Introductory</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/introductory_workflow_french_version.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">26</span>&nbsp; <span class="chapter-title">Introductory (French)</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/introductory_workflow_dutch_version.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">27</span>&nbsp; <span class="chapter-title">Introductory (Dutch)</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-11" role="navigation" aria-expanded="false">
 <span class="menu-text">Training</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-11" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-11" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/training.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">28</span>&nbsp; <span class="chapter-title">Training</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/exercises.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">29</span>&nbsp; <span class="chapter-title">Exercises</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-12" role="navigation" aria-expanded="false">
 <span class="menu-text">Support &amp; resources</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-12" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-12" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/support.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">30</span>&nbsp; <span class="chapter-title">Support</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/resources.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">31</span>&nbsp; <span class="chapter-title">Resources</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/acknowledgments.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">32</span>&nbsp; <span class="chapter-title">Contributions</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-13" role="navigation" aria-expanded="false">
 <span class="menu-text">Appendices</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-13" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-13" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/msea.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">A</span>&nbsp; <span class="chapter-title">Microbe Set Enrichment Analysis (MSEA)</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/mmuphin_meta_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">B</span>&nbsp; <span class="chapter-title">Meta-analyses</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/extra_material.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">C</span>&nbsp; <span class="chapter-title">Extra material</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/visualization.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">D</span>&nbsp; <span class="chapter-title">Visualization</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/session_info.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">E</span>&nbsp; <span class="chapter-title">Technical info</span></span></a>
  </div>
</li>
      </ul>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">Table of contents</h2>
   
  <ul>
<li>
<a href="#sec-loading-experimental-microbiome-data" id="toc-sec-loading-experimental-microbiome-data" class="nav-link active" data-scroll-target="#sec-loading-experimental-microbiome-data"><span class="header-section-number">4.1</span> Import microbiome data from files</a>
  <ul class="collapse">
<li><a href="#standard-data-formats" id="toc-standard-data-formats" class="nav-link" data-scroll-target="#standard-data-formats"><span class="header-section-number">4.1.1</span> Standard data formats</a></li>
  <li><a href="#sec-import-from-file" id="toc-sec-import-from-file" class="nav-link" data-scroll-target="#sec-import-from-file"><span class="header-section-number">4.1.2</span> Non-standard formats</a></li>
  </ul>
</li>
  <li>
<a href="#sec-example-data" id="toc-sec-example-data" class="nav-link" data-scroll-target="#sec-example-data"><span class="header-section-number">4.2</span> Data resources</a>
  <ul class="collapse">
<li><a href="#sec-package-data" id="toc-sec-package-data" class="nav-link" data-scroll-target="#sec-package-data"><span class="header-section-number">4.2.1</span> Package data</a></li>
  <li><a href="#experimenthub-data" id="toc-experimenthub-data" class="nav-link" data-scroll-target="#experimenthub-data"><span class="header-section-number">4.2.2</span> ExperimentHub data</a></li>
  <li><a href="#curated-metagenomic-data" id="toc-curated-metagenomic-data" class="nav-link" data-scroll-target="#curated-metagenomic-data"><span class="header-section-number">4.2.3</span> Curated metagenomic data</a></li>
  <li><a href="#human-microbiome-compendium" id="toc-human-microbiome-compendium" class="nav-link" data-scroll-target="#human-microbiome-compendium"><span class="header-section-number">4.2.4</span> Human microbiome compendium</a></li>
  <li><a href="#other-data-sources" id="toc-other-data-sources" class="nav-link" data-scroll-target="#other-data-sources"><span class="header-section-number">4.2.5</span> Other data sources</a></li>
  </ul>
</li>
  </ul></nav>
    </div>
<!-- main -->
<main class="content page-columns page-full" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../pages/containers.html">Data containers &amp; importing</a></li><li class="breadcrumb-item"><a href="../pages/import.html"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Import</span></a></li></ol></nav><div class="quarto-title">
<h1 class="title"><span id="sec-import" class="quarto-section-identifier"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Import</span></span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header><div class="cell" data-layout-align="center">
<script>
document.addEventListener("click", function (event) {
    if (event.target.classList.contains("rebook-collapse")) {
        event.target.classList.toggle("active");
        var content = event.target.nextElementSibling;
        if (content.style.display === "block") {
            content.style.display = "none";
        } else {
            content.style.display = "block";
        }
    }
})
</script><style>
.rebook-collapse {
  background-color: #eee;
  color: #444;
  cursor: pointer;
  padding: 18px;
  width: 100%;
  border: none;
  text-align: left;
  outline: none;
  font-size: 15px;
}

.rebook-content {
  padding: 0 18px;
  display: none;
  overflow: hidden;
  background-color: #f1f1f1;
}
</style>
</div>
<p>This chapter demonstrates how data can be imported into data containers from files. Additionally, multiple databases offer curated microbiome data, which are introduced in <a href="#sec-example-data" class="quarto-xref"><span>Section 4.2</span></a>.</p>
<p>In prior to importing the data to container, the raw sequences must be mapped to abundances. The specific method depends on sequencing type and other data-specific factors. There are several resources available to guide this process <a href="https://www.metagenomics.wiki/">Metagenomics wiki</a> being one of them. In <a href="extra_material.html#sec-16s-workflow" class="quarto-xref"><span>Section C.2</span></a> we provide an example workflow for processing 16S rRNA sequences into a data container using DADA2.</p>
<section id="sec-loading-experimental-microbiome-data" class="level2" data-number="4.1"><h2 data-number="4.1" class="anchored" data-anchor-id="sec-loading-experimental-microbiome-data">
<span class="header-section-number">4.1</span> Import microbiome data from files</h2>
<p>The data containers can be constructed from scratch. For most common microbiome data formats, there is dedicated importers available which streamlines the importing.</p>
<section id="standard-data-formats" class="level3" data-number="4.1.1"><h3 data-number="4.1.1" class="anchored" data-anchor-id="standard-data-formats">
<span class="header-section-number">4.1.1</span> Standard data formats</h3>
<p>Specific import functions are provided for:</p>
<ul>
<li>BIOM files (see <code>help(mia::importBIOM)</code>)</li>
<li>QIIME2 files (see <code>help(mia::importQIIME2)</code>)</li>
<li>Mothur files (see <code>help(mia::importMothur)</code>)</li>
<li>MetaPhlAn files (see <code>help(mia::importMetaPhlAn)</code>)</li>
<li>HUMAnN files (see <code>help(mia::importHUMAnN)</code>)</li>
<li>taxpasta files (see <code>help(mia::importTaxpasta)</code>)</li>
</ul>
<p>Here we show how <a href="https://biom-format.org/">Biom files</a> are imported into a <code>TreeSE</code> object using as an example Tengeler2020, which is further described <a href="https://microbiome.github.io/mia/reference/Tengeler2020.html">here</a>. This dataset consists of 3 files, which can be fetched or downloaded from <a href="https://github.com/microbiome/data/tree/main/Tengeler2020">this repository</a>:</p>
<ul>
<li>biom file: abundance table and taxonomy information</li>
<li>csv file: sample metadata</li>
<li>tree file: phylogenetic tree</li>
</ul>
<p>To begin with, we store the data in a local directory within the working directory, such as <em>data/</em>, and define the source file paths.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">biom_file_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span></span>
<span>    <span class="st">"extdata"</span>, <span class="st">"Aggregated_humanization2.biom"</span>, package <span class="op">=</span> <span class="st">"OMA"</span><span class="op">)</span></span>
<span><span class="va">sample_meta_file_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span></span>
<span>    <span class="st">"extdata"</span>, <span class="st">"Mapping_file_ADHD_aggregated.csv"</span>, package <span class="op">=</span> <span class="st">"OMA"</span><span class="op">)</span></span>
<span><span class="va">tree_file_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span></span>
<span>    <span class="st">"extdata"</span>, <span class="st">"Data_humanization_phylo_aggregation.tre"</span>, package <span class="op">=</span> <span class="st">"OMA"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now we can read in the biom file and convert it into a <code>TreeSE</code> object. In addition, we retrieve the rank names from the prefixes of the feature names and then remove them with the <code>rank.from.prefix</code> and <code>prefix.rm</code> optional arguments.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/microbiome/mia">mia</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># read biom and convert it to TreeSE</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/importBIOM.html">importBIOM</a></span><span class="op">(</span></span>
<span>    <span class="va">biom_file_path</span>,</span>
<span>    rank.from.prefix <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    prefix.rm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    artifact.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Check</span></span>
<span><span class="va">tse</span></span>
<span><span class="co">##  class: TreeSummarizedExperiment </span></span>
<span><span class="co">##  dim: 151 27 </span></span>
<span><span class="co">##  metadata(0):</span></span>
<span><span class="co">##  assays(1): counts</span></span>
<span><span class="co">##  rownames(151): 1726470 1726471 ... 17264756 17264757</span></span>
<span><span class="co">##  rowData names(6): kingdom phylum ... family genus</span></span>
<span><span class="co">##  colnames(27): A110 A111 ... A38 A39</span></span>
<span><span class="co">##  colData names(0):</span></span>
<span><span class="co">##  reducedDimNames(0):</span></span>
<span><span class="co">##  mainExpName: NULL</span></span>
<span><span class="co">##  altExpNames(0):</span></span>
<span><span class="co">##  rowLinks: NULL</span></span>
<span><span class="co">##  rowTree: NULL</span></span>
<span><span class="co">##  colLinks: NULL</span></span>
<span><span class="co">##  colTree: NULL</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The <code>assays</code> slot includes a list of abundance tables. The imported abundance table is named as “counts”. Let us inspect only the first cols and rows.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">tse</span>, <span class="st">"counts"</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span>
<span><span class="co">##            A110  A111  A12</span></span>
<span><span class="co">##  1726470  17722 11630    0</span></span>
<span><span class="co">##  1726471  12052     0 2679</span></span>
<span><span class="co">##  17264731     0   970    0</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The <code>rowdata</code> includes taxonomic information from the biom file. The <code><a href="https://rdrr.io/r/utils/head.html">head()</a></code> command shows just the beginning of the data table for an overview.</p>
<p><code><a href="https://rdrr.io/pkg/knitr/man/kable.html">knitr::kable()</a></code> helps print the information more nicely.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">##  DataFrame with 6 rows and 6 columns</span></span>
<span><span class="co">##               kingdom          phylum            class              order</span></span>
<span><span class="co">##           &lt;character&gt;     &lt;character&gt;      &lt;character&gt;        &lt;character&gt;</span></span>
<span><span class="co">##  1726470     Bacteria   Bacteroidetes      Bacteroidia      Bacteroidales</span></span>
<span><span class="co">##  1726471     Bacteria   Bacteroidetes      Bacteroidia      Bacteroidales</span></span>
<span><span class="co">##  17264731    Bacteria   Bacteroidetes      Bacteroidia      Bacteroidales</span></span>
<span><span class="co">##  17264726    Bacteria   Bacteroidetes      Bacteroidia      Bacteroidales</span></span>
<span><span class="co">##  1726472     Bacteria Verrucomicrobia Verrucomicrobiae Verrucomicrobiales</span></span>
<span><span class="co">##  17264724    Bacteria   Bacteroidetes      Bacteroidia      Bacteroidales</span></span>
<span><span class="co">##                        family           genus</span></span>
<span><span class="co">##                   &lt;character&gt;     &lt;character&gt;</span></span>
<span><span class="co">##  1726470       Bacteroidaceae     Bacteroides</span></span>
<span><span class="co">##  1726471       Bacteroidaceae     Bacteroides</span></span>
<span><span class="co">##  17264731  Porphyromonadaceae Parabacteroides</span></span>
<span><span class="co">##  17264726      Bacteroidaceae     Bacteroides</span></span>
<span><span class="co">##  1726472  Verrucomicrobiaceae     Akkermansia</span></span>
<span><span class="co">##  17264724      Bacteroidaceae     Bacteroides</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We notice that the imported biom file did not contain any <code>colData</code> yet, so only an empty dataframe appears in this slot.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">##  DataFrame with 6 rows and 0 columns</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Let us add <code>colData</code> from the sample metadata, which is stored in a CSV file.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># CSV file with colnames in the first row and rownames in the first column</span></span>
<span><span class="va">sample_meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span></span>
<span>    <span class="va">sample_meta_file_path</span>, sep <span class="op">=</span> <span class="st">","</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add this sample data to colData of the taxonomic data object</span></span>
<span><span class="co"># Note that the data must be given in a DataFrame format</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html">DataFrame</a></span><span class="op">(</span><span class="va">sample_meta</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now the <code>colData</code> includes the sample metadata.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">##  DataFrame with 6 rows and 4 columns</span></span>
<span><span class="co">##         Treatment      Cohort TreatmentxCohort Description</span></span>
<span><span class="co">##       &lt;character&gt; &lt;character&gt;      &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">##  A110        ADHD    Cohort_1    ADHD_Cohort_1        A110</span></span>
<span><span class="co">##  A12         ADHD    Cohort_1    ADHD_Cohort_1         A12</span></span>
<span><span class="co">##  A15         ADHD    Cohort_1    ADHD_Cohort_1         A15</span></span>
<span><span class="co">##  A19         ADHD    Cohort_1    ADHD_Cohort_1         A19</span></span>
<span><span class="co">##  A21         ADHD    Cohort_2    ADHD_Cohort_2         A21</span></span>
<span><span class="co">##  A23         ADHD    Cohort_2    ADHD_Cohort_2         A23</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Finally, we add a phylogenetic tree to the <code>rowData</code> slot. Such feature is available only in <code>TreeSE</code> objects. Similarly, Trees specifying the sample hierarchy can be stored in the <code>colTree</code> slot.</p>
<p>Here, we read in the file containing the phylogenetic tree and insert it in corresponding slot of the <code>TreeSE</code> object.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Reads the tree file</span></span>
<span><span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/read.tree.html">read.tree</a></span><span class="op">(</span><span class="va">tree_file_path</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add tree to rowTree</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/TreeSummarizedExperiment-accessor.html">rowTree</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">tree</span></span>
<span></span>
<span><span class="co"># Check</span></span>
<span><span class="va">tse</span></span>
<span><span class="co">##  class: TreeSummarizedExperiment </span></span>
<span><span class="co">##  dim: 151 27 </span></span>
<span><span class="co">##  metadata(0):</span></span>
<span><span class="co">##  assays(1): counts</span></span>
<span><span class="co">##  rownames(151): 1726470 1726471 ... 17264756 17264757</span></span>
<span><span class="co">##  rowData names(6): kingdom phylum ... family genus</span></span>
<span><span class="co">##  colnames(27): A110 A12 ... A35 A38</span></span>
<span><span class="co">##  colData names(4): Treatment Cohort TreatmentxCohort Description</span></span>
<span><span class="co">##  reducedDimNames(0):</span></span>
<span><span class="co">##  mainExpName: NULL</span></span>
<span><span class="co">##  altExpNames(0):</span></span>
<span><span class="co">##  rowLinks: a LinkDataFrame (151 rows)</span></span>
<span><span class="co">##  rowTree: 1 phylo tree(s) (151 leaves)</span></span>
<span><span class="co">##  colLinks: NULL</span></span>
<span><span class="co">##  colTree: NULL</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now the <code>rowTree</code> slot contains the phylogenetic tree:</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/TreeSummarizedExperiment-accessor.html">rowTree</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="sec-import-from-file" class="level3" data-number="4.1.2"><h3 data-number="4.1.2" class="anchored" data-anchor-id="sec-import-from-file">
<span class="header-section-number">4.1.2</span> Non-standard formats</h3>
<p>Microbiome (taxonomic) profiling data is commonly distributed in various file formats. You can import such external data files as a <code>TreeSE</code> object, but the details depend on the file format. Here, we provide examples for common formats. Some datasets and raw files to learn how to import raw data and construct <code>TreeSE/MAE</code> containers are available in the <a href="https://github.com/microbiome/data">microbiome data repository</a>.</p>
<section id="csv-import" class="level4" data-number="4.1.2.1"><h4 data-number="4.1.2.1" class="anchored" data-anchor-id="csv-import">
<span class="header-section-number">4.1.2.1</span> CSV import</h4>
<p><strong>CSV data tables</strong> can be imported with the standard R functions, then converted to the desired format. For detailed examples, you can check the <a href="https://bioconductor.org/help/course-materials/2019/BSS2019/04_Practical_CoreApproachesInBioconductor.html">Bioconductor course material</a> by Martin Morgan. You can also check the <a href="https://github.com/microbiome/OMA/tree/master/data">example files</a> and construct your own CSV files accordingly.</p>
<p>Recommendations for the CSV files are the following. File names are arbitrary; we refer here to the same names as in the examples:</p>
<ul>
<li><p>Abundance table (<code>assay_taxa.csv</code>): data matrix (features x samples); first column provides feature IDs, the first row provides sample IDs; other values should be numeric (abundances).</p></li>
<li><p>Row data (<code>rowdata_taxa.csv</code>): data table (features x info); first column provides feature IDs, the first row provides column headers; this file usually contains the taxonomic mapping between different taxonomic levels. Ideally, the feature IDs (row names) match one-to-one with the abundance table row names.</p></li>
<li><p>Column data (<code>coldata.csv</code>): data table (samples x info); first column provides sample IDs, the first row provides column headers; this file usually contains the sample metadata/phenodata (such as subject age, health etc). Ideally, the sample IDs match one-to-one with the abundance table column names.</p></li>
</ul>
<p>After you have set up the CSV files, you can read them in R:</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">count_file</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"assay_taxa.csv"</span>, package <span class="op">=</span> <span class="st">"OMA"</span><span class="op">)</span></span>
<span><span class="va">tax_file</span>    <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"rowdata_taxa.csv"</span>, package <span class="op">=</span> <span class="st">"OMA"</span><span class="op">)</span></span>
<span><span class="va">sample_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"coldata.csv"</span>, package <span class="op">=</span> <span class="st">"OMA"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load files</span></span>
<span><span class="va">counts</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span><span class="va">count_file</span>, row.names<span class="op">=</span><span class="fl">1</span><span class="op">)</span>   <span class="co"># Abundance table (e.g. ASV data; to assay data)</span></span>
<span><span class="va">tax</span>     <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span><span class="va">tax_file</span>, row.names<span class="op">=</span><span class="fl">1</span><span class="op">)</span>     <span class="co"># Taxonomy table (to rowData)</span></span>
<span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span><span class="va">sample_file</span>, row.names<span class="op">=</span><span class="fl">1</span><span class="op">)</span>  <span class="co"># Sample data (to colData)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>After reading the data in R, ensure the following:</p>
<ul>
<li><p>abundance table (<code>counts</code>): numeric <code>matrix</code>, with feature IDs as rownames and sample IDs as column names.</p></li>
<li><p><code>rowdata</code> (<code>tax</code>): <code>DataFrame</code>, with feature IDs as rownames. If this is a <code>data.frame</code> you can use the function <code><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html">DataFrame()</a></code> to change the format. Column names are free but in microbiome analysis they usually they refer to taxonomic ranks. The rownames in rowdata should match with rownames in abundance table.</p></li>
<li><p><code>coldata</code> (<code>samples</code>): <code>DataFrame</code>, with sample IDs as rownames. If this is a <code>data.frame</code> you can use the function <code><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html">DataFrame()</a></code> to change the format. Column names are free. The rownames in <code>coldata</code> should match with colnames in abundance table.</p></li>
</ul>
<p><strong>Always ensure that the tables have rownames!</strong> The <em>TreeSE</em> constructor compares rownames and ensures that, for example, right samples are linked with right patient.</p>
<p>Also, ensure that the row and column names match one-to-one between abundance table, <code>rowdata</code>, and <code>coldata</code>:</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Match rows and columns</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">counts</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">tax</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">samples</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Let's ensure that the data is in correct (numeric matrix) format:</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>If you hesitate about the format of the data, you can compare to one of the available demonstration datasets, and make sure that your data components have the same format.</p>
<p>There are many different source files and many different ways to read data in R. One can do data manipulation in R as well. Investigate the entries as follows.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># coldata rownames match assay colnames</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/all.html">all</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">samples</span><span class="op">)</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span> <span class="co"># our dataset</span></span>
<span><span class="co">##  [1] TRUE</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">samples</span><span class="op">)</span> <span class="co"># should be data.frame or DataFrame</span></span>
<span><span class="co">##  [1] "data.frame"</span></span>
<span></span>
<span><span class="co"># rowdata rownames match assay rownames</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/all.html">all</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">tax</span><span class="op">)</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span> <span class="co"># our dataset</span></span>
<span><span class="co">##  [1] TRUE</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">tax</span><span class="op">)</span> <span class="co"># should be data.frame or DataFrame</span></span>
<span><span class="co">##  [1] "data.frame"</span></span>
<span></span>
<span><span class="co"># Counts</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="co"># should be a numeric matrix</span></span>
<span><span class="co">##  [1] "matrix" "array"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Important!
</div>
</div>
<div class="callout-body-container callout-body">
<p>Ensure that colnames of assay match with rownames of <code>colData</code>, and rownames of assay match with rownames of <code>rowData</code>.</p>
<p>If your data do not have names, you have to be especially careful, since this can lead to errors!</p>
</div>
</div>
</section><section id="constructing-treese" class="level4" data-number="4.1.2.2"><h4 data-number="4.1.2.2" class="anchored" data-anchor-id="constructing-treese">
<span class="header-section-number">4.1.2.2</span> Constructing TreeSE</h4>
<p>Now, let’s create the <code>TreeSE</code> object from the input data tables. Here we also convert the data objects in their preferred formats:</p>
<ul>
<li>counts –&gt; numeric matrix</li>
<li>rowData –&gt; DataFrame</li>
<li>colData –&gt; DataFrame</li>
</ul>
<p>The <code>SimpleList</code> could be used to include multiple alternative assays, if necessary.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Create a TreeSE</span></span>
<span><span class="va">tse_taxa</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/TreeSummarizedExperiment-constructor.html">TreeSummarizedExperiment</a></span><span class="op">(</span></span>
<span>    assays <span class="op">=</span>  <span class="fu">SimpleList</span><span class="op">(</span>counts <span class="op">=</span> <span class="va">counts</span><span class="op">)</span>,</span>
<span>    colData <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html">DataFrame</a></span><span class="op">(</span><span class="va">samples</span><span class="op">)</span>,</span>
<span>    rowData <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html">DataFrame</a></span><span class="op">(</span><span class="va">tax</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tse_taxa</span></span>
<span><span class="co">##  class: TreeSummarizedExperiment </span></span>
<span><span class="co">##  dim: 12706 40 </span></span>
<span><span class="co">##  metadata(0):</span></span>
<span><span class="co">##  assays(1): counts</span></span>
<span><span class="co">##  rownames(12706): GAYR01026362.62.2014 CVJT01000011.50.2173 ...</span></span>
<span><span class="co">##    JRJTB:03787:02429 JRJTB:03787:02478</span></span>
<span><span class="co">##  rowData names(7): Phylum Class ... Species OTU</span></span>
<span><span class="co">##  colnames(40): C1 C2 ... C39 C40</span></span>
<span><span class="co">##  colData names(6): Sample Rat ... Fat XOS</span></span>
<span><span class="co">##  reducedDimNames(0):</span></span>
<span><span class="co">##  mainExpName: NULL</span></span>
<span><span class="co">##  altExpNames(0):</span></span>
<span><span class="co">##  rowLinks: NULL</span></span>
<span><span class="co">##  rowTree: NULL</span></span>
<span><span class="co">##  colLinks: NULL</span></span>
<span><span class="co">##  colTree: NULL</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now you should have a ready-made <code>TreeSE</code> data object that can be used in downstream analyses.</p>
</section><section id="constructing-mae" class="level4" data-number="4.1.2.3"><h4 data-number="4.1.2.3" class="anchored" data-anchor-id="constructing-mae">
<span class="header-section-number">4.1.2.3</span> Constructing MAE</h4>
<p>To construct a <code>MAE</code> object, just combine multiple <code>TreeSE</code> data containers.</p>
<p>Here we import metabolite data from the same study.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">count_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"assay_metabolites.csv"</span>, package <span class="op">=</span> <span class="st">"OMA"</span><span class="op">)</span></span>
<span><span class="va">sample_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"coldata.csv"</span>, package <span class="op">=</span> <span class="st">"OMA"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load files</span></span>
<span><span class="va">counts</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span><span class="va">count_file</span>, row.names<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span><span class="va">sample_file</span>, row.names<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create a TreeSE for the metabolite data</span></span>
<span><span class="va">tse_metabolite</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/TreeSummarizedExperiment-constructor.html">TreeSummarizedExperiment</a></span><span class="op">(</span></span>
<span>    assays <span class="op">=</span> <span class="fu">SimpleList</span><span class="op">(</span>concs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    colData <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html">DataFrame</a></span><span class="op">(</span><span class="va">samples</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tse_metabolite</span></span>
<span><span class="co">##  class: TreeSummarizedExperiment </span></span>
<span><span class="co">##  dim: 38 40 </span></span>
<span><span class="co">##  metadata(0):</span></span>
<span><span class="co">##  assays(1): concs</span></span>
<span><span class="co">##  rownames(38): Butyrate Acetate ... Malonate 1,3-dihydroxyacetone</span></span>
<span><span class="co">##  rowData names(0):</span></span>
<span><span class="co">##  colnames(40): C1 C2 ... C39 C40</span></span>
<span><span class="co">##  colData names(6): Sample Rat ... Fat XOS</span></span>
<span><span class="co">##  reducedDimNames(0):</span></span>
<span><span class="co">##  mainExpName: NULL</span></span>
<span><span class="co">##  altExpNames(0):</span></span>
<span><span class="co">##  rowLinks: NULL</span></span>
<span><span class="co">##  rowTree: NULL</span></span>
<span><span class="co">##  colLinks: NULL</span></span>
<span><span class="co">##  colTree: NULL</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Important!
</div>
</div>
<div class="callout-body-container callout-body">
<p>When creating TreeSE, <code>assay</code> must be a matrix, and both <code>colData</code> and <code>rowData</code> must be DataFrame objects.</p>
</div>
</div>
<p>Now we can combine these two experiments into <em>MAE</em>.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Create an ExperimentList that includes experiments</span></span>
<span><span class="va">experiments</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://github.com/waldronlab/MultiAssayExperiment/reference/ExperimentList.html">ExperimentList</a></span><span class="op">(</span></span>
<span>    microbiome <span class="op">=</span> <span class="va">tse_taxa</span>, metabolite <span class="op">=</span> <span class="va">tse_metabolite</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create a MAE</span></span>
<span><span class="va">mae</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://github.com/waldronlab/MultiAssayExperiment/reference/MultiAssayExperiment.html">MultiAssayExperiment</a></span><span class="op">(</span>experiments <span class="op">=</span> <span class="va">experiments</span><span class="op">)</span></span>
<span></span>
<span><span class="va">mae</span></span>
<span><span class="co">##  A MultiAssayExperiment object of 2 listed</span></span>
<span><span class="co">##   experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##   Containing an ExperimentList class object of length 2:</span></span>
<span><span class="co">##   [1] microbiome: TreeSummarizedExperiment with 12706 rows and 40 columns</span></span>
<span><span class="co">##   [2] metabolite: TreeSummarizedExperiment with 38 rows and 40 columns</span></span>
<span><span class="co">##  Functionality:</span></span>
<span><span class="co">##   experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##   colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##   sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##   `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##   *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##   assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##   exportClass() - save data to flat files</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section></section></section><section id="sec-example-data" class="level2 page-columns page-full" data-number="4.2"><h2 data-number="4.2" class="anchored" data-anchor-id="sec-example-data">
<span class="header-section-number">4.2</span> Data resources</h2>
<p>Open demonstration data for testing and benchmarking purposes is available from multiple locations. This chapter introduces some options. The other chapters of this book provide ample examples about the use of the data.</p>
<section id="sec-package-data" class="level3" data-number="4.2.1"><h3 data-number="4.2.1" class="anchored" data-anchor-id="sec-package-data">
<span class="header-section-number">4.2.1</span> Package data</h3>
<p>The <code>mia</code> R package contains example datasets that are direct conversions from the alternative <code>phyloseq</code> container to the <code>TreeSE</code> container.</p>
<p>List the <a href="https://microbiome.github.io/mia/reference/mia-datasets.html">available datasets</a> in the <code>mia</code> package:</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/microbiome/mia">mia</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"mia"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Load the <code>GlobalPatterns</code> data from the <code>mia</code> package:</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"GlobalPatterns"</span>, package<span class="op">=</span><span class="st">"mia"</span><span class="op">)</span></span>
<span><span class="va">GlobalPatterns</span></span>
<span><span class="co">##  class: TreeSummarizedExperiment </span></span>
<span><span class="co">##  dim: 19216 26 </span></span>
<span><span class="co">##  metadata(0):</span></span>
<span><span class="co">##  assays(1): counts</span></span>
<span><span class="co">##  rownames(19216): 549322 522457 ... 200359 271582</span></span>
<span><span class="co">##  rowData names(7): Kingdom Phylum ... Genus Species</span></span>
<span><span class="co">##  colnames(26): CL3 CC1 ... Even2 Even3</span></span>
<span><span class="co">##  colData names(7): X.SampleID Primer ... SampleType Description</span></span>
<span><span class="co">##  reducedDimNames(0):</span></span>
<span><span class="co">##  mainExpName: NULL</span></span>
<span><span class="co">##  altExpNames(0):</span></span>
<span><span class="co">##  rowLinks: a LinkDataFrame (19216 rows)</span></span>
<span><span class="co">##  rowTree: 1 phylo tree(s) (19216 leaves)</span></span>
<span><span class="co">##  colLinks: NULL</span></span>
<span><span class="co">##  colTree: NULL</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>R packages contain additional demonstration data sets (see the Datasets section of the reference page):</p>
<ul>
<li><a href="https://microbiome.github.io/mia/reference/index.html">mia reference</a></li>
<li><a href="https://microbiome.github.io/miaViz/reference/index.html">miaViz reference</a></li>
<li><a href="https://microbiome.github.io/miaTime/reference/index.html">miaTime reference</a></li>
</ul></section><section id="experimenthub-data" class="level3 page-columns page-full" data-number="4.2.2"><h3 data-number="4.2.2" class="anchored" data-anchor-id="experimenthub-data">
<span class="header-section-number">4.2.2</span> ExperimentHub data</h3>
<p><a href="https://bioconductor.org/packages/release/bioc/vignettes/ExperimentHub/inst/doc/ExperimentHub.html">ExperimentHub</a> provides a variety of data resources, including the <a href="https://bioconductor.org/packages/release/data/experiment/html/microbiomeDataSets.html">microbiomeDataSets</a> package <span class="citation" data-cites="Morgan2021 microlahti2021">(<a href="#ref-Morgan2021" role="doc-biblioref">Morgan and Shepherd 2021</a>; <a href="#ref-microlahti2021" role="doc-biblioref">Lahti, Ernst, and Shetty 2021</a>)</span>.</p>
<div class="no-row-height column-margin column-container"><div id="ref-Morgan2021" class="csl-entry" role="listitem">
Morgan, Martin, and Lori Shepherd. 2021. <em>ExperimentHub: Client to Access ExperimentHub Resources</em>.
</div><div id="ref-microlahti2021" class="csl-entry" role="listitem">
Lahti, Leo, Felix G. M. Ernst, and Sudarshan Shetty. 2021. <em>microbiomeDataSets: Experiment Hub Based Microbiome Datasets</em>.
</div></div><p>A table of the available datasets is available through the <code><a href="https://rdrr.io/pkg/microbiomeDataSets/man/availableDataSets.html">availableDataSets()</a></code> function.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">microbiomeDataSets</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/microbiomeDataSets/man/availableDataSets.html">availableDataSets</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">##              Dataset</span></span>
<span><span class="co">##  1  GrieneisenTSData</span></span>
<span><span class="co">##  2       LahtiMLData</span></span>
<span><span class="co">##  3        LahtiMData</span></span>
<span><span class="co">##  4      OKeefeDSData</span></span>
<span><span class="co">##  5 SilvermanAGutData</span></span>
<span><span class="co">##  6        SongQAData</span></span>
<span><span class="co">##  7   SprockettTHData</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>All data are downloaded from ExperimentHub and cached for local re-use. Check the <a href="https://microbiome.github.io/microbiomeDataSets/reference/index.html">man pages of each function</a> for a detailed documentation of the data contents and references. Let us retrieve a <em><a href="https://bioconductor.org/packages/3.20/MAE">MAE</a></em> dataset:</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># mae &lt;- HintikkaXOData()</span></span>
<span><span class="co"># Since HintikkaXOData is now added to mia, we can load it directly from there</span></span>
<span><span class="co"># We suggest to check other datasets from microbiomeDataSets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">HintikkaXOData</span>, package <span class="op">=</span> <span class="st">"mia"</span><span class="op">)</span></span>
<span><span class="va">mae</span> <span class="op">&lt;-</span> <span class="va">HintikkaXOData</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Data is available in <em><a href="https://bioconductor.org/packages/3.20/SE">SE</a></em>, <em><a href="https://bioconductor.org/packages/3.20/TreeSE">TreeSE</a></em> and <em><a href="https://bioconductor.org/packages/3.20/MAE">MAE</a></em> data containers; see the for example <a href="cross_correlation.html#sec-cross-correlation" class="quarto-xref"><span>Section 20.1</span></a> for more details.</p>
</section><section id="curated-metagenomic-data" class="level3 page-columns page-full" data-number="4.2.3"><h3 data-number="4.2.3" class="anchored" data-anchor-id="curated-metagenomic-data">
<span class="header-section-number">4.2.3</span> Curated metagenomic data</h3>
<p><a href="https://bioconductor.org/packages/release/data/experiment/html/curatedMetagenomicData.html">curatedMetagenomicData</a> is a large collection of curated human microbiome datasets, provided as <code>TreeSE</code> objects <span class="citation" data-cites="Pasolli2017">(<a href="#ref-Pasolli2017" role="doc-biblioref">Pasolli et al. 2017</a>)</span>. The resource provides curated human microbiome data including gene families, marker abundance, marker presence, pathway abundance, pathway coverage, and relative abundance for samples from different body sites. See the package homepage for more details on data availability and access.</p>
<div class="no-row-height column-margin column-container"><div id="ref-Pasolli2017" class="csl-entry" role="listitem">
Pasolli, Edoardo, Lucas Schiffer, Paolo Manghi, Audrey Renson, Valerie Obenchain, Duy Tin Truong, Francesco Beghini, et al. 2017. <span>“Accessible, Curated Metagenomic Data Through <span>ExperimentHub</span>.”</span> <em>Nature Methods</em> 14 (11): 1023–24. https://doi.org/<a href="https://doi.org/10.1038/nmeth.4468">https://doi.org/10.1038/nmeth.4468</a>.
</div><div id="ref-Vatanen2016" class="csl-entry" role="listitem">
Vatanen, Tommi, Aleksandar D. Kostic, Eva d’Hennezel, Heli Siljander, Eric A. Franzosa, Moran Yassour, Raivo Kolde, et al. 2016. <span>“Variation in Microbiome LPS Immunogenicity Contributes to Autoimmunity in Humans.”</span> <em>Cell</em> 165 (May): 842–53. <a href="https://doi.org/10.1016/j.cell.2016.04.007">https://doi.org/10.1016/j.cell.2016.04.007</a>.
</div></div><p>As one example, let us retrieve the Vatanen (2016) <span class="citation" data-cites="Vatanen2016">(<a href="#ref-Vatanen2016" role="doc-biblioref">Vatanen et al. 2016</a>)</span> data set. This is a larger collection with a bit longer download time.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/waldronlab/curatedMetagenomicData">curatedMetagenomicData</a></span><span class="op">)</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/curatedMetagenomicData/man/curatedMetagenomicData.html">curatedMetagenomicData</a></span><span class="op">(</span><span class="st">"Vatanen*"</span>, dryrun <span class="op">=</span> <span class="cn">FALSE</span>, counts <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="human-microbiome-compendium" class="level3 page-columns page-full" data-number="4.2.4"><h3 data-number="4.2.4" class="anchored" data-anchor-id="human-microbiome-compendium">
<span class="header-section-number">4.2.4</span> Human microbiome compendium</h3>
<p><a href="https://seandavi.github.io/MicroBioMap/">MicroBioMap</a> dataset includes over 170k samples of publicly available 16S rRNA amplicon sequencing data, all processed using the same pipeline and reference database <span class="citation" data-cites="microbiomap">(<a href="#ref-microbiomap" role="doc-biblioref">Davis et al. 2024</a>)</span>. After installing the MicroBioMap package (see the <a href="https://github.com/seandavi/MicroBioMap#microbiome-compendium">original website</a> for instructions), you can load the compendium with</p>
<div class="no-row-height column-margin column-container"><div id="ref-microbiomap" class="csl-entry" role="listitem">
Davis, Sean, Richard Abdill, Ran Blehkman, Samantha Graham, and Casey Greene. 2024. <em>MicroBioMap: Access the Microbiome Compendium from r</em>. <a href="https://github.com/seandavi/MicroBioMap">https://github.com/seandavi/MicroBioMap</a>.
</div></div><div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">MicroBioMap</span><span class="op">)</span></span>
<span><span class="va">cpd</span> <span class="op">&lt;-</span> <span class="fu">getCompendium</span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>This returns a <code>TreeSE</code> object. Currently, the <code>rowTree</code> slot of the <code>TreeSE</code> is not populated.</p>
<p>After loading the compendium, you will have immediate access to nearly 170,000 microbiome samples of publicly available 16S rRNA amplicon sequencing data, all processed using the same pipeline and reference database. For more use examples in R/Bioconductor, see the <a href="https://seandavi.github.io/MicroBioMap/articles/overview.html">MicroBioMap vignette</a>.</p>
</section><section id="other-data-sources" class="level3" data-number="4.2.5"><h3 data-number="4.2.5" class="anchored" data-anchor-id="other-data-sources">
<span class="header-section-number">4.2.5</span> Other data sources</h3>
<p>The current collections provide access to vast microbiome data resources. The output has to be converted into <code>TreeSE/MAE</code> separately.</p>
<ul>
<li>
<a href="https://bioconductor.org/packages/release/bioc/html/MGnifyR.html">MGnifyR</a> provides access to <a href="https://www.ebi.ac.uk/metagenomics/">EBI/MGnify</a>
</li>
<li>
<a href="https://bioconductor.org/packages/release/bioc/html/HoloFoodR.html">HoloFoodR</a> provides access to <a href="https://www.holofooddata.org/">EBI/HoloFood</a>
</li>
<li>
<a href="https://github.com/cran/qiitr">qiitr</a> provides access to <a href="https://qiita.com/about">QIITA</a>
</li>
<li>
<a href="https://github.com/microbiome/qiime2R">qiime2R</a> provides access to <a href="https://docs.qiime2.org/2024.2/">QIIME2</a>
</li>
</ul>



</section></section><a onclick="window.scrollTo(0, 0); return false;" role="button" id="quarto-back-to-top"><i class="bi bi-arrow-up"></i> Back to top</a></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = false;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="../pages/containers.html" class="pagination-link" aria-label="Data containers">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Data containers</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../pages/convert.html" class="pagination-link" aria-label="Convert &amp; export">
        <span class="nav-page-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Convert &amp; export</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
<p><a href="./">Orchestrating Microbiome Analysis with Bioconductor</a></p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">
<p>This book was built with <a href="https://github.com/js2264/BiocBook/">BiocBook</a> with <span class="emoji" data-emoji="heart">❤️</span></p>
</div>
  </div>
</footer>


</body></html>