<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>26&nbsp; Introductory (French) ‚Äì Orchestrating Microbiome Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../pages/introductory_workflow_dutch_version.html" rel="next">
<link href="../pages/introductory_workflow.html" rel="prev">
<link href="../assets/favicon.png" rel="icon" type="image/png">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script src="../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><link href="../site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="../site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script><link href="../site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="../site_libs/datatables-binding-0.33/datatables.js"></script><script src="../site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><link href="../site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="../site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="../site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script><link href="../site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="../site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script><script src="../site_libs/plotly-binding-4.10.4/plotly.js"></script><script src="../site_libs/typedarray-0.1/typedarray.min.js"></script><link href="../site_libs/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet">
<script src="../site_libs/plotly-main-2.11.1/plotly-latest.min.js"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script><script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>
</head>
<body class="nav-sidebar docked nav-fixed slimcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../assets/mia_logo.png" alt="" class="navbar-logo"></a>
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Orchestrating Microbiome Analysis</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-bi-github">
<li>
    <a class="dropdown-item" href="https://github.com/microbiome/OMA/">
 <span class="dropdown-text">Source Code</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://microbiome.github.io/OMA/docs/devel/">
 <span class="dropdown-text">Browse version <code>devel</code></span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com"> <i class="bi bi-twitter" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
</div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../pages/introductory_workflow.html">Workflows</a></li><li class="breadcrumb-item"><a href="../pages/introductory_workflow_french_version.html"><span class="chapter-number">26</span>&nbsp; <span class="chapter-title">Introductory (French)</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto"><div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Overview</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false">
 <span class="menu-text">Introduction</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Microbiome data science in Bioconductor</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/miaverse.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">miaverse</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false">
 <span class="menu-text">Data containers &amp; importing</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/containers.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Data containers</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/import.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Import</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/convert.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Convert &amp; export</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false">
 <span class="menu-text">Data wrangling</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/taxonomy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Taxonomic information</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/wrangling.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Data wrangling</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/subsetting.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Subsetting</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/agglomeration.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Agglomeration</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/transformation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Transformation</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false">
 <span class="menu-text">Exploration &amp; QC</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/quality_control.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Exploration &amp; quality control</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/composition.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Community composition</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false">
 <span class="menu-text">Diversity &amp; similarity</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/alpha_diversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Community diversity</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/beta_diversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Community similarity</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/clustering.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">Community typing</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" role="navigation" aria-expanded="false">
 <span class="menu-text">Association</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/differential_abundance.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">16</span>&nbsp; <span class="chapter-title">Differential abundance</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/correlation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">17</span>&nbsp; <span class="chapter-title">Correlation</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" role="navigation" aria-expanded="false">
 <span class="menu-text">Networks</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-7" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/network_learning.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">18</span>&nbsp; <span class="chapter-title">Network learning &amp; analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/network_comparison.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">19</span>&nbsp; <span class="chapter-title">Network comparison</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" role="navigation" aria-expanded="false">
 <span class="menu-text">Multiomics</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-8" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/cross_correlation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">20</span>&nbsp; <span class="chapter-title">Cross-association</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/multiassay_ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">21</span>&nbsp; <span class="chapter-title">Ordination-based multiassay analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/integrated_learner.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">22</span>&nbsp; <span class="chapter-title">Multi-omics prediction and classification</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-9" role="navigation" aria-expanded="false">
 <span class="menu-text">Machine learning &amp; statistical modeling</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-9" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-9" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/machine_learning.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">23</span>&nbsp; <span class="chapter-title">Machine learning</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/statistical_modeling.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">24</span>&nbsp; <span class="chapter-title">Statistical modeling</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-10" role="navigation" aria-expanded="true">
 <span class="menu-text">Workflows</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-10" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-10" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/introductory_workflow.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">25</span>&nbsp; <span class="chapter-title">Introductory</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/introductory_workflow_french_version.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">26</span>&nbsp; <span class="chapter-title">Introductory (French)</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/introductory_workflow_dutch_version.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">27</span>&nbsp; <span class="chapter-title">Introductory (Dutch)</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-11" role="navigation" aria-expanded="false">
 <span class="menu-text">Training</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-11" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-11" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/training.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">28</span>&nbsp; <span class="chapter-title">Training</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/exercises.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">29</span>&nbsp; <span class="chapter-title">Exercises</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-12" role="navigation" aria-expanded="false">
 <span class="menu-text">Support &amp; resources</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-12" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-12" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/support.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">30</span>&nbsp; <span class="chapter-title">Support</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/resources.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">31</span>&nbsp; <span class="chapter-title">Resources</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/acknowledgments.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">32</span>&nbsp; <span class="chapter-title">Contributions</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-13" role="navigation" aria-expanded="false">
 <span class="menu-text">Appendices</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-13" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-13" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/msea.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">A</span>&nbsp; <span class="chapter-title">Microbe Set Enrichment Analysis (MSEA)</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/mmuphin_meta_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">B</span>&nbsp; <span class="chapter-title">Meta-analyses</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/extra_material.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">C</span>&nbsp; <span class="chapter-title">Extra material</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/visualization.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">D</span>&nbsp; <span class="chapter-title">Visualization</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/session_info.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">E</span>&nbsp; <span class="chapter-title">Technical info</span></span></a>
  </div>
</li>
      </ul>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
        
    </div>
<!-- main -->
<main class="content page-columns page-full" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../pages/introductory_workflow.html">Workflows</a></li><li class="breadcrumb-item"><a href="../pages/introductory_workflow_french_version.html"><span class="chapter-number">26</span>&nbsp; <span class="chapter-title">Introductory (French)</span></a></li></ol></nav><div class="quarto-title">
<h1 class="title">
<span class="chapter-number">26</span>&nbsp; <span class="chapter-title">Introductory (French)</span>
</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header><p><strong>Version:</strong> 1.0</p>
<div class="cell" data-layout-align="center">
<script>
document.addEventListener("click", function (event) {
    if (event.target.classList.contains("rebook-collapse")) {
        event.target.classList.toggle("active");
        var content = event.target.nextElementSibling;
        if (content.style.display === "block") {
            content.style.display = "none";
        } else {
            content.style.display = "block";
        }
    }
})
</script><style>
.rebook-collapse {
  background-color: #eee;
  color: #444;
  cursor: pointer;
  padding: 18px;
  width: 100%;
  border: none;
  text-align: left;
  outline: none;
  font-size: 15px;
}

.rebook-content {
  padding: 0 18px;
  display: none;
  overflow: hidden;
  background-color: #f1f1f1;
}
</style>
</div>
<div id="workflow" class="page-columns page-full">
<div id="translate">
<p><a href="introductory_workflow.html"><img src="figures/flags/GB_flag.png" alt="Great Britain flag" class="flag"><strong>English</strong> </a> <a href="introductory_workflow_dutch_version.html"><img src="figures/flags/Netherland_flag.jpg" alt="Netherland flag" class="flag"><strong>Dutch</strong> </a></p>
</div>
<p><img src="figures/intro_workflow_logo.svg" alt="logo workflow" id="banner"></p>
<section id="introduction" class="level2" data-number="26.1"><h2 data-number="26.1" class="anchored" data-anchor-id="introduction">
<span class="header-section-number">26.1</span> Introduction</h2>
<p>Bonjour et bienvenue dans un workflow complet utilisant les derniers outils R/Bioconductor pour la science des donn√©es du microbiome. Dans ce tutoriel, nous vous guiderons √† travers quelques √©tapes de base d‚Äôune √©tude d‚Äôanalyse de composition utilisant OMA. Celles-ci seront applicables √† presque tous vos projets et vous aideront √† comprendre les concepts fondamentaux qui propulseront üöÄ vos futures analyses du microbiome.</p>
</section><section id="importation-des-donn√©es" class="level2 page-columns page-full" data-number="26.2"><h2 data-number="26.2" class="anchored" data-anchor-id="importation-des-donn√©es">
<span class="header-section-number">26.2</span> Importation des donn√©es</h2>
<p>Lors de l‚Äôutilisation de packages pour le microbiome, il existe de nombreuses fa√ßons diff√©rentes d‚Äôimporter vos donn√©es. Commen√ßons par charger les packages requis:</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># List of packages that we need</span></span>
<span><span class="va">packages</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"ggplot2"</span>, <span class="st">"knitr"</span>, <span class="st">"mia"</span>, <span class="st">"dplyr"</span>, <span class="st">"miaViz"</span>, <span class="st">"vegan"</span>, <span class="st">"DT"</span>,</span>
<span>    <span class="st">"scater"</span>, <span class="st">"patchwork"</span>, <span class="st">"sechm"</span>, <span class="st">"plotly"</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span><span class="co"># Get packages that are already installed installed</span></span>
<span><span class="va">packages_already_installed</span> <span class="op">&lt;-</span> <span class="va">packages</span><span class="op">[</span> <span class="va">packages</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html">installed.packages</a></span><span class="op">(</span><span class="op">)</span> <span class="op">]</span></span>
<span></span>
<span><span class="co"># Get packages that need to be installed</span></span>
<span><span class="va">packages_need_to_install</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html">setdiff</a></span><span class="op">(</span> <span class="va">packages</span>, <span class="va">packages_already_installed</span> <span class="op">)</span></span>
<span></span>
<span><span class="co"># Loads BiocManager into the session. Install it if it not already installed.</span></span>
<span><span class="kw">if</span><span class="op">(</span> <span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/">"BiocManager"</a></span><span class="op">)</span> <span class="op">)</span><span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/">"BiocManager"</a></span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># If there are packages that need to be installed, installs them with BiocManager</span></span>
<span><span class="co"># Updates old packages.</span></span>
<span><span class="kw">if</span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">packages_need_to_install</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span> <span class="op">)</span> <span class="op">{</span></span>
<span>   <span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html">install</a></span><span class="op">(</span><span class="va">packages_need_to_install</span>, ask <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Load all packages into session. Stop if there are packages that were not</span></span>
<span><span class="co"># successfully loaded</span></span>
<span><span class="va">pkgs_not_loaded</span> <span class="op">&lt;-</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">packages</span>, <span class="va">require</span>, character.only <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">pkgs_not_loaded</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">pkgs_not_loaded</span><span class="op">)</span><span class="op">[</span> <span class="va">pkgs_not_loaded</span> <span class="op">]</span></span>
<span><span class="kw">if</span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">pkgs_not_loaded</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span> <span class="op">)</span><span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/stop.html">stop</a></span><span class="op">(</span></span>
<span>        <span class="st">"Error in loading the following packages into the session: '"</span>,</span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">pkgs_not_loaded</span>, collapse <span class="op">=</span> <span class="st">"', '"</span><span class="op">)</span>, <span class="st">"'"</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Vous pouvez choisir d‚Äôutiliser vos propres donn√©es ou l‚Äôun des ensembles de donn√©es int√©gr√©s fournis par mia que vous trouverez ici <a href="import.html#sec-example-data" class="quarto-xref"><span>Section 4.2</span></a>:</p>
<p>Dans ce tutoriel, nous utiliserons l‚Äôensemble de donn√©es <span class="citation" data-cites="Tengeler2020">C Tengeler et al. (<a href="#ref-Tengeler2020" role="doc-biblioref">2020</a>)</span>. Cet ensemble de donn√©es a √©t√© cr√©√© par A.C. Tengeler pour essayer de d√©montrer l‚Äôimpact des microbiomes alt√©r√©s sur la structure du cerveau. Voici comment nous pouvons charger les donn√©es dans notre environnement R :</p>
<div class="no-row-height column-margin column-container"><div id="ref-Tengeler2020" class="csl-entry" role="listitem">
C Tengeler, Anouk, Sarita A Dam, Maximilian Wiesmann, Jilly Naaijen, Miranda van Bodegom, Clara Belzer, Pieter J Dederen, et al. 2020. <span>‚ÄúGut Microbiota from Persons with Attention-Deficit/Hyperactivity Disorder Affects the Brain in Mice.‚Äù</span> <em>Microbiome</em> 8: 1‚Äì14. <a href="https://doi.org/10.1186/s40168-020-00816-x">https://doi.org/10.1186/s40168-020-00816-x</a>.
</div></div><div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"Tengeler2020"</span>, package<span class="op">=</span><span class="st">"mia"</span><span class="op">)</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="va">Tengeler2020</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Bien s√ªr, il existe d‚Äôautres moyens d‚Äôimporter vos donn√©es en utilisant le <a href="https://microbiome.github.io/mia/"><strong>package mia</strong></a>. Ceux-ci incluent : utiliser <strong>vos propres donn√©es</strong> (<a href="import.html#sec-import-from-file" class="quarto-xref"><span>Section 4.1.2</span></a>) ou convertir un objet existant en un objet <code>TreeSummarizedExperiment</code> comme indiqu√© dans <strong>cette section</strong> : <a href="convert.html#sec-conversions-between-data-formats-in-r" class="quarto-xref"><span>Section 5.1</span></a>.</p>
<hr></section><section id="stockage-des-donn√©es-du-microbiome" class="level2 page-columns page-full" data-number="26.3"><h2 data-number="26.3" class="anchored" data-anchor-id="stockage-des-donn√©es-du-microbiome">
<span class="header-section-number">26.3</span> Stockage des donn√©es du microbiome</h2>
<p><code>TreeSummarizedExperiment</code> ou objet TreeSE est le type d‚Äôobjet utilis√© dans le <a href="https://microbiome.github.io/mia/"><strong>package mia</strong></a> pour stocker vos donn√©es. C‚Äôest un type de donn√©es polyvalent et multi-usages qui permet de stocker et d‚Äôacc√©der aux donn√©es de mani√®re efficace.</p>
<p>Voici un rappel rapide sur la fa√ßon d‚Äôacc√©der √† certains types de donn√©es :</p>
<p>Vous pouvez acc√©der aux assays <strong>assays</strong> (<a href="containers.html#sec-assay-slot" class="quarto-xref"><span>Section 3.3</span></a>) de cette mani√®re :</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">assay</span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span></span>
<span><span class="co">##                   A110   A12  A15  A19  A21  A23  A25  A28 A29  A34</span></span>
<span><span class="co">##  Bacteroides     17722 11630    0 8806 1740 1791 2368 1316 252 5702</span></span>
<span><span class="co">##  Bacteroides_1   12052     0 2679 2776  540  229    0    0   0 6347</span></span>
<span><span class="co">##  Parabacteroides     0   970    0  549  145    0  109  119  31    0</span></span>
<span><span class="co">##  Bacteroides_2       0  1911    0 5497  659    0  588  542 141    0</span></span>
<span><span class="co">##  Akkermansia      1143  1891 1212  584   84  700  440  244  25 1611</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Alors que <code>colData</code> (<a href="wrangling.html#sec-add-or-modify-data" class="quarto-xref"><span>Section 7.2</span></a>) est accessible via :</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Transform the colData to a dataframe</span></span>
<span><span class="va">tse_df_colData</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Show as an interactive table</span></span>
<span><span class="fu">datatable</span><span class="op">(</span><span class="va">tse_df_colData</span>,options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>pageLength <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>,rownames <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="datatables html-widget html-fill-item" id="htmlwidget-2f8a5fb6d21e4e1eb5ff" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-2f8a5fb6d21e4e1eb5ff">{"x":{"filter":"none","vertical":false,"data":[["ADHD","ADHD","ADHD","ADHD","ADHD","ADHD","ADHD","ADHD","ADHD","ADHD","ADHD","ADHD","ADHD","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control"],["Cohort_1","Cohort_1","Cohort_1","Cohort_1","Cohort_2","Cohort_2","Cohort_2","Cohort_2","Cohort_2","Cohort_3","Cohort_3","Cohort_3","Cohort_3","Cohort_1","Cohort_1","Cohort_1","Cohort_1","Cohort_1","Cohort_1","Cohort_2","Cohort_2","Cohort_2","Cohort_2","Cohort_2","Cohort_3","Cohort_3","Cohort_3"],["ADHD_Cohort_1","ADHD_Cohort_1","ADHD_Cohort_1","ADHD_Cohort_1","ADHD_Cohort_2","ADHD_Cohort_2","ADHD_Cohort_2","ADHD_Cohort_2","ADHD_Cohort_2","ADHD_Cohort_3","ADHD_Cohort_3","ADHD_Cohort_3","ADHD_Cohort_3","Control_Cohort_1","Control_Cohort_1","Control_Cohort_1","Control_Cohort_1","Control_Cohort_1","Control_Cohort_1","Control_Cohort_2","Control_Cohort_2","Control_Cohort_2","Control_Cohort_2","Control_Cohort_2","Control_Cohort_3","Control_Cohort_3","Control_Cohort_3"],["A110","A12","A15","A19","A21","A23","A25","A28","A29","A34","A36","A37","A39","A111","A13","A14","A16","A17","A18","A210","A22","A24","A26","A27","A33","A35","A38"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>patient_status<\/th>\n      <th>cohort<\/th>\n      <th>patient_status_vs_cohort<\/th>\n      <th>sample_name<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"name":"patient_status","targets":0},{"name":"cohort","targets":1},{"name":"patient_status_vs_cohort","targets":2},{"name":"sample_name","targets":3}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p><strong>rowData</strong> (<a href="containers.html#sec-rowData" class="quarto-xref"><span>Section 3.5</span></a>) contient des donn√©es sur les caract√©ristiques des √©chantillons, notamment des informations taxonomiques.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tse_df_rowData</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">datatable</span><span class="op">(</span><span class="va">tse_df_rowData</span>, options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>pageLength <span class="op">=</span> <span class="fl">5</span><span class="op">)</span><span class="op">)</span>  </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="datatables html-widget html-fill-item" id="htmlwidget-757496678b3e660328db" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-757496678b3e660328db">{"x":{"filter":"none","vertical":false,"data":[["Bacteroides","Bacteroides_1","Parabacteroides","Bacteroides_2","Akkermansia","Bacteroides_3","Parabacteroides_1","Bacteroides_4","Bacteroides_5","Parabacteroides_2","Escherichia-Shigella","Subdoligranulum","Lachnoclostridium","[Ruminococcus]_gnavus_group","[Ruminococcus]_gnavus_group_1","Faecalibacterium","Ruminococcus_1","Ruminococcaceae_UCG-013","[Clostridium]_innocuum_group","Eubacterium","Lachnoclostridium_1","Subdoligranulum_1","Unidentified_Lachnospiraceae","Bacteroides_6","Unidentified_Lachnospiraceae_1","Bacteroides_7","[Clostridium]_innocuum_group_1","Parabacteroides_3","Hungatella","[Eubacterium]_rectale_group","Hungatella_1","Alistipes","Paraprevotella","Blautia","Lachnospiraceae_ND3007_group","Hungatella_2","Escherichia-Shigella_1","Roseburia","Hungatella_3","Erysipelatoclostridium","Alistipes_1","Faecalibacterium_1","Bacteroides_8","Subdoligranulum_2","Unidentified_Lachnospiraceae_2","Unidentified_Lachnospiraceae_3","Alistipes_2","Blautia_1","[Ruminococcus]_gauvreauii_group","Unidentified_Lachnospiraceae_4","Ruminococcaceae_UCG-013_1","Bacteroides_9","Blautia_2","Blautia_3","Barnesiella","Faecalibacterium_2","Unidentified_Lachnospiraceae_5","Parabacteroides_4","Bacteroides_10","Eisenbergiella","Eubacterium_1","Hungatella_4","Blautia_4","Hungatella_5","Blautia_5","Subdoligranulum_3","[Ruminococcus]_gauvreauii_group_1","[Eubacterium]_xylanophilum_group","Unidentified_Ruminococcaceae","Alistipes_3","Coprobacter","Escherichia-Shigella_2","Unidentified_Bacteroidales","[Eubacterium]_fissicatena_group","Anaerostipes","Ruminococcaceae_UCG-013_2","Unidentified_Lachnospiraceae_6","Bacteroides_11","Lachnoclostridium_2","Bacteroides_12","Paraprevotella_1","Alistipes_4","Odoribacter","Dielma","[Ruminococcus]_gnavus_group_2","Ruminococcaceae_UCG-013_3","Coprococcus_2","Subdoligranulum_4","Ruminococcus_2","Unidentified_Lachnospiraceae_7","Bacteroides_13","Unidentified_Lachnospiraceae_8","Ruminiclostridium_5","Flavonifractor","Fusicatenibacter","Blautia_6","Holdemania","Hungatella_6","Bacteroides_14","Hungatella_7","Unidentified_Lachnospiraceae_9","Clostridium_sensu_stricto_1","Subdoligranulum_5","Alistipes_5","Epulopiscium","[Ruminococcus]_gauvreauii_group_2","Fusicatenibacter_1","Unidentified_Lachnospiraceae_10","Anaerotruncus","Parabacteroides_5","Barnesiella_1","Unidentified_Porphyromonadaceae","Ruminococcaceae_UCG-004","Alistipes_6","Faecalibacterium_3","Bacteroides_15","Escherichia-Shigella_3","Ruminococcaceae_UCG-014","Unidentified_Clostridiales","Anaerostipes_1","Blautia_7","Unidentified_Lachnospiraceae_11","Unidentified_Lachnospiraceae_12","Lachnospiraceae_ND3007_group_1","Unidentified_Lachnospiraceae_13","uncultured_bacterium","Butyricicoccus","Akkermansia_1","Parabacteroides_6","[Ruminococcus]_gauvreauii_group_3","Unidentified_Enterobacteriaceae","Ruminiclostridium_9","Hungatella_8","Escherichia-Shigella_4","[Eubacterium]_coprostanoligenes_group","Parabacteroides_7","Catabacter","Roseburia_1","Bacteroides_16","Lachnoclostridium_3","Ruminococcaceae_UCG-013_4","Unidentified_Ruminococcaceae_1","Subdoligranulum_6","Catabacter_1","Bacteroides_17","Bacteroides_18","Unidentified_Gastranaerophilales","Halomonas","Lachnoclostridium_4","Parabacteroides_8","Unidentified_Lachnospiraceae_14"],["Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria"],["Bacteroidetes","Bacteroidetes","Bacteroidetes","Bacteroidetes","Verrucomicrobia","Bacteroidetes","Bacteroidetes","Bacteroidetes","Bacteroidetes","Bacteroidetes","Proteobacteria","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Bacteroidetes","Firmicutes","Bacteroidetes","Firmicutes","Bacteroidetes","Firmicutes","Firmicutes","Firmicutes","Bacteroidetes","Bacteroidetes","Firmicutes","Firmicutes","Firmicutes","Proteobacteria","Firmicutes","Firmicutes","Firmicutes","Bacteroidetes","Firmicutes","Bacteroidetes","Firmicutes","Firmicutes","Firmicutes","Bacteroidetes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Bacteroidetes","Firmicutes","Firmicutes","Bacteroidetes","Firmicutes","Firmicutes","Bacteroidetes","Bacteroidetes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Bacteroidetes","Bacteroidetes","Proteobacteria","Bacteroidetes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Bacteroidetes","Firmicutes","Bacteroidetes","Bacteroidetes","Bacteroidetes","Bacteroidetes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Bacteroidetes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Bacteroidetes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Bacteroidetes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Bacteroidetes","Bacteroidetes","Bacteroidetes","Firmicutes","Bacteroidetes","Firmicutes","Bacteroidetes","Proteobacteria","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Bacteroidetes","Firmicutes","Verrucomicrobia","Bacteroidetes","Firmicutes","Proteobacteria","Firmicutes","Firmicutes","Proteobacteria","Firmicutes","Bacteroidetes","Firmicutes","Firmicutes","Bacteroidetes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Bacteroidetes","Bacteroidetes","Cyanobacteria","Proteobacteria","Firmicutes","Bacteroidetes","Firmicutes"],["Bacteroidia","Bacteroidia","Bacteroidia","Bacteroidia","Verrucomicrobiae","Bacteroidia","Bacteroidia","Bacteroidia","Bacteroidia","Bacteroidia","Gammaproteobacteria","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Erysipelotrichia","Clostridia","Clostridia","Clostridia","Clostridia","Bacteroidia","Clostridia","Bacteroidia","Erysipelotrichia","Bacteroidia","Clostridia","Clostridia","Clostridia","Bacteroidia","Bacteroidia","Clostridia","Clostridia","Clostridia","Gammaproteobacteria","Clostridia","Clostridia","Erysipelotrichia","Bacteroidia","Clostridia","Bacteroidia","Clostridia","Clostridia","Clostridia","Bacteroidia","Clostridia","Clostridia","Clostridia","Clostridia","Bacteroidia","Clostridia","Clostridia","Bacteroidia","Clostridia","Clostridia","Bacteroidia","Bacteroidia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Bacteroidia","Bacteroidia","Gammaproteobacteria","Bacteroidia","Clostridia","Clostridia","Clostridia","Clostridia","Bacteroidia","Clostridia","Bacteroidia","Bacteroidia","Bacteroidia","Bacteroidia","Erysipelotrichia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Bacteroidia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Erysipelotrichia","Clostridia","Bacteroidia","Clostridia","Clostridia","Clostridia","Clostridia","Bacteroidia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Bacteroidia","Bacteroidia","Bacteroidia","Clostridia","Bacteroidia","Clostridia","Bacteroidia","Gammaproteobacteria","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Bacteroidia","Clostridia","Verrucomicrobiae","Bacteroidia","Clostridia","Gammaproteobacteria","Clostridia","Clostridia","Gammaproteobacteria","Clostridia","Bacteroidia","Clostridia","Clostridia","Bacteroidia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Bacteroidia","Bacteroidia","Melainabacteria","Gammaproteobacteria","Clostridia","Bacteroidia","Clostridia"],["Bacteroidales","Bacteroidales","Bacteroidales","Bacteroidales","Verrucomicrobiales","Bacteroidales","Bacteroidales","Bacteroidales","Bacteroidales","Bacteroidales","Enterobacteriales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Erysipelotrichales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Bacteroidales","Clostridiales","Bacteroidales","Erysipelotrichales","Bacteroidales","Clostridiales","Clostridiales","Clostridiales","Bacteroidales","Bacteroidales","Clostridiales","Clostridiales","Clostridiales","Enterobacteriales","Clostridiales","Clostridiales","Erysipelotrichales","Bacteroidales","Clostridiales","Bacteroidales","Clostridiales","Clostridiales","Clostridiales","Bacteroidales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Bacteroidales","Clostridiales","Clostridiales","Bacteroidales","Clostridiales","Clostridiales","Bacteroidales","Bacteroidales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Bacteroidales","Bacteroidales","Enterobacteriales","Bacteroidales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Bacteroidales","Clostridiales","Bacteroidales","Bacteroidales","Bacteroidales","Bacteroidales","Erysipelotrichales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Bacteroidales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Erysipelotrichales","Clostridiales","Bacteroidales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Bacteroidales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Bacteroidales","Bacteroidales","Bacteroidales","Clostridiales","Bacteroidales","Clostridiales","Bacteroidales","Enterobacteriales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Bacteroidales","Clostridiales","Verrucomicrobiales","Bacteroidales","Clostridiales","Enterobacteriales","Clostridiales","Clostridiales","Enterobacteriales","Clostridiales","Bacteroidales","Clostridiales","Clostridiales","Bacteroidales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Bacteroidales","Bacteroidales","Gastranaerophilales","Oceanospirillales","Clostridiales","Bacteroidales","Clostridiales"],["Bacteroidaceae","Bacteroidaceae","Porphyromonadaceae","Bacteroidaceae","Verrucomicrobiaceae","Bacteroidaceae","Porphyromonadaceae","Bacteroidaceae","Bacteroidaceae","Porphyromonadaceae","Enterobacteriaceae","Ruminococcaceae","Lachnospiraceae","Lachnospiraceae","Lachnospiraceae","Ruminococcaceae","Ruminococcaceae","Ruminococcaceae","Erysipelotrichaceae","Eubacteriaceae","Lachnospiraceae","Ruminococcaceae","Lachnospiraceae","Bacteroidaceae","Lachnospiraceae","Bacteroidaceae","Erysipelotrichaceae","Porphyromonadaceae","Lachnospiraceae","Lachnospiraceae","Lachnospiraceae","Rikenellaceae","Prevotellaceae","Lachnospiraceae","Lachnospiraceae","Lachnospiraceae","Enterobacteriaceae","Lachnospiraceae","Lachnospiraceae","Erysipelotrichaceae","Rikenellaceae","Ruminococcaceae","Bacteroidaceae","Ruminococcaceae","Lachnospiraceae","Lachnospiraceae","Rikenellaceae","Lachnospiraceae","Lachnospiraceae","Lachnospiraceae","Ruminococcaceae","Bacteroidaceae","Lachnospiraceae","Lachnospiraceae","Porphyromonadaceae","Ruminococcaceae","Lachnospiraceae","Porphyromonadaceae","Bacteroidaceae","Lachnospiraceae","Eubacteriaceae","Lachnospiraceae","Lachnospiraceae","Lachnospiraceae","Lachnospiraceae","Ruminococcaceae","Lachnospiraceae","Lachnospiraceae","Ruminococcaceae","Rikenellaceae","Porphyromonadaceae","Enterobacteriaceae","","Lachnospiraceae","Lachnospiraceae","Ruminococcaceae","Lachnospiraceae","Bacteroidaceae","Lachnospiraceae","Bacteroidaceae","Prevotellaceae","Rikenellaceae","Porphyromonadaceae","Erysipelotrichaceae","Lachnospiraceae","Ruminococcaceae","Lachnospiraceae","Ruminococcaceae","Ruminococcaceae","Lachnospiraceae","Bacteroidaceae","Lachnospiraceae","Ruminococcaceae","Ruminococcaceae","Lachnospiraceae","Lachnospiraceae","Erysipelotrichaceae","Lachnospiraceae","Bacteroidaceae","Lachnospiraceae","Lachnospiraceae","Clostridiaceae_1","Ruminococcaceae","Rikenellaceae","Lachnospiraceae","Lachnospiraceae","Lachnospiraceae","Lachnospiraceae","Ruminococcaceae","Porphyromonadaceae","Porphyromonadaceae","Porphyromonadaceae","Ruminococcaceae","Rikenellaceae","Ruminococcaceae","Bacteroidaceae","Enterobacteriaceae","Ruminococcaceae","","Lachnospiraceae","Lachnospiraceae","Lachnospiraceae","Lachnospiraceae","Lachnospiraceae","Lachnospiraceae","Bacteroidales_S24-7_group","Ruminococcaceae","Verrucomicrobiaceae","Porphyromonadaceae","Lachnospiraceae","Enterobacteriaceae","Ruminococcaceae","Lachnospiraceae","Enterobacteriaceae","Ruminococcaceae","Porphyromonadaceae","Christensenellaceae","Lachnospiraceae","Bacteroidaceae","Lachnospiraceae","Ruminococcaceae","Ruminococcaceae","Ruminococcaceae","Christensenellaceae","Bacteroidaceae","Bacteroidaceae","","Halomonadaceae","Lachnospiraceae","Porphyromonadaceae","Lachnospiraceae"],["Bacteroides","Bacteroides","Parabacteroides","Bacteroides","Akkermansia","Bacteroides","Parabacteroides","Bacteroides","Bacteroides","Parabacteroides","Escherichia-Shigella","Subdoligranulum","Lachnoclostridium","[Ruminococcus]_gnavus_group","[Ruminococcus]_gnavus_group","Faecalibacterium","Ruminococcus_1","Ruminococcaceae_UCG-013","[Clostridium]_innocuum_group","Eubacterium","Lachnoclostridium","Subdoligranulum","","Bacteroides","","Bacteroides","[Clostridium]_innocuum_group","Parabacteroides","Hungatella","[Eubacterium]_rectale_group","Hungatella","Alistipes","Paraprevotella","Blautia","Lachnospiraceae_ND3007_group","Hungatella","Escherichia-Shigella","Roseburia","Hungatella","Erysipelatoclostridium","Alistipes","Faecalibacterium","Bacteroides","Subdoligranulum","","uncultured","Alistipes","Blautia","[Ruminococcus]_gauvreauii_group","","Ruminococcaceae_UCG-013","Bacteroides","Blautia","Blautia","Barnesiella","Faecalibacterium","","Parabacteroides","Bacteroides","Eisenbergiella","Eubacterium","Hungatella","Blautia","Hungatella","Blautia","Subdoligranulum","[Ruminococcus]_gauvreauii_group","[Eubacterium]_xylanophilum_group","uncultured","Alistipes","Coprobacter","Escherichia-Shigella","","[Eubacterium]_fissicatena_group","Anaerostipes","Ruminococcaceae_UCG-013","","Bacteroides","Lachnoclostridium","Bacteroides","Paraprevotella","Alistipes","Odoribacter","Dielma","[Ruminococcus]_gnavus_group","Ruminococcaceae_UCG-013","Coprococcus_2","Subdoligranulum","Ruminococcus_2","","Bacteroides","","Ruminiclostridium_5","Flavonifractor","Fusicatenibacter","Blautia","Holdemania","Hungatella","Bacteroides","Hungatella","","Clostridium_sensu_stricto_1","Subdoligranulum","Alistipes","Epulopiscium","[Ruminococcus]_gauvreauii_group","Fusicatenibacter","","Anaerotruncus","Parabacteroides","Barnesiella","uncultured","Ruminococcaceae_UCG-004","Alistipes","Faecalibacterium","Bacteroides","Escherichia-Shigella","Ruminococcaceae_UCG-014","","Anaerostipes","Blautia","","","Lachnospiraceae_ND3007_group","","uncultured_bacterium","Butyricicoccus","Akkermansia","Parabacteroides","[Ruminococcus]_gauvreauii_group","","Ruminiclostridium_9","Hungatella","Escherichia-Shigella","[Eubacterium]_coprostanoligenes_group","Parabacteroides","Catabacter","Roseburia","Bacteroides","Lachnoclostridium","Ruminococcaceae_UCG-013","uncultured","Subdoligranulum","Catabacter","Bacteroides","Bacteroides","","Halomonas","Lachnoclostridium","Parabacteroides","uncultured"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Kingdom<\/th>\n      <th>Phylum<\/th>\n      <th>Class<\/th>\n      <th>Order<\/th>\n      <th>Family<\/th>\n      <th>Genus<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Kingdom","targets":1},{"name":"Phylum","targets":2},{"name":"Class","targets":3},{"name":"Order","targets":4},{"name":"Family","targets":5},{"name":"Genus","targets":6}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p>Ici <code>rowData(tse)</code> renvoie un DataFrame avec 151 lignes et 7 colonnes. Chaque ligne repr√©sente un organisme et chaque colonne un niveau taxonomique.</p>
<p>Pour illustrer la structure d‚Äôun <code>TreeSummarizedExperiment</code>, voici un article de <span class="citation" data-cites="Huang2021">Huang et al. (<a href="#ref-Huang2021" role="doc-biblioref">2021</a>)</span> qui utilise ce type d‚Äôobjet. De plus, veuillez consulter la figure 1 ci-dessous.</p>
<div class="no-row-height column-margin column-container"><div id="ref-Huang2021" class="csl-entry" role="listitem">
Huang, Ruizhu, Charlotte Soneson, Felix G. M. Ernst, et al. 2021. <span>‚ÄúTreeSummarizedExperiment: A S4 Class for Data with Hierarchical Structure [Version 2; Peer Review: 3 Approved].‚Äù</span> <em>F1000Research</em> 9: 1246. <a href="https://doi.org/10.12688/f1000research.26669.2">https://doi.org/10.12688/f1000research.26669.2</a>.
</div></div><div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><a href="figures/TSE-structure.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-1" title="1. Storing microbial data: the structure of a TreeSummarizedExperiment"><img src="figures/TSE-structure.svg" class="contentimg img-fluid figure-img" alt="1. Storing microbial data: the structure of a TreeSummarizedExperiment"></a></p>
<figcaption>1. Storing microbial data: the structure of a TreeSummarizedExperiment</figcaption></figure>
</div>
<hr></section><section id="manipulation-des-donn√©es" class="level2" data-number="26.4"><h2 data-number="26.4" class="anchored" data-anchor-id="manipulation-des-donn√©es">
<span class="header-section-number">26.4</span> Manipulation des donn√©es</h2>
<p>Dans certains cas, vous devrez peut-√™tre modifier vos donn√©es pour obtenir les r√©sultats souhait√©s. Dans cette section, nous verrons comment agglom√©rer les donn√©es, les sous-ensembles et plus encore. Un <code>TreeSummarizedExperiment</code> permet une manipulation astucieuse des donn√©es en utilisant le <a href="https://dplyr.tidyverse.org/"><strong>package dplyr</strong></a>.</p>
<section id="sous-ensembles" class="level3" data-number="26.4.1"><h3 data-number="26.4.1" class="anchored" data-anchor-id="sous-ensembles">
<span class="header-section-number">26.4.1</span> Sous-ensembles</h3>
<p>Dans certains cas, vous n‚Äôaurez peut-√™tre besoin d‚Äôutiliser qu‚Äôune partie de votre <code>TreeSummarizedExperiment</code> d‚Äôorigine.</p>
<p>En utilisant l‚Äôensemble de donn√©es Tengeler2020, nous pouvons nous concentrer sur une certaine cohorte par exemple. Cela est assez simple :</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tse_subset_by_sample</span> <span class="op">&lt;-</span> <span class="va">tse</span><span class="op">[</span> , <span class="va">tse</span><span class="op">$</span><span class="va">cohort</span> <span class="op">==</span><span class="st">"Cohort_1"</span><span class="op">]</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Cela cr√©era un objet <code>TreeSummarizedExperiment</code> ne contenant que les √©chantillons de la premi√®re cohorte.</p>
</section><section id="agglom√©ration-des-donn√©es" class="level3" data-number="26.4.2"><h3 data-number="26.4.2" class="anchored" data-anchor-id="agglom√©ration-des-donn√©es">
<span class="header-section-number">26.4.2</span> Agglom√©ration des donn√©es</h3>
<p>Pour pousser davantage votre analyse de donn√©es et vous concentrer sur sa distribution √† un rang taxonomique sp√©cifique, il peut √™tre b√©n√©fique d‚Äôagglom√©rer vos donn√©es √† ce niveau particulier. La fonction <code>agglomerateByRank()</code> simplifie ce processus, permettant des analyses plus fluides et efficaces. Voici un exemple :</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tse.agglomerated</span> <span class="op">&lt;-</span> <span class="fu">agglomerateByRank</span><span class="op">(</span><span class="va">tse</span>, rank<span class="op">=</span><span class="st">'Phylum'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Check</span></span>
<span><span class="fu">datatable</span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">tse.agglomerated</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>pageLength <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>,rownames <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="datatables html-widget html-fill-item" id="htmlwidget-70908ed2339f2dc41196" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-70908ed2339f2dc41196">{"x":{"filter":"none","vertical":false,"data":[["Bacteria","Bacteria","Bacteria","Bacteria","Bacteria"],["Bacteroidetes","Cyanobacteria","Firmicutes","Proteobacteria","Verrucomicrobia"],[null,null,null,null,null],[null,null,null,null,null],[null,null,null,null,null],[null,null,null,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Kingdom<\/th>\n      <th>Phylum<\/th>\n      <th>Class<\/th>\n      <th>Order<\/th>\n      <th>Family<\/th>\n      <th>Genus<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"name":"Kingdom","targets":0},{"name":"Phylum","targets":1},{"name":"Class","targets":2},{"name":"Order","targets":3},{"name":"Family","targets":4},{"name":"Genus","targets":5}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p>G√©nial ! Maintenant, nos donn√©es sont confin√©es aux informations taxonomiques jusqu‚Äôau niveau du Phylum, permettant √† l‚Äôanalyse de se concentrer sur ce rang sp√©cifique. Dans le reste du workflow, nous n‚Äôutiliserons pas les donn√©es agglom√©r√©es, mais tout le code ci-dessous peut √™tre utilis√© sur celles-ci.</p>
</section></section><section id="indicateurs" class="level2 page-columns page-full" data-number="26.5"><h2 data-number="26.5" class="anchored" data-anchor-id="indicateurs">
<span class="header-section-number">26.5</span> Indicateurs</h2>
<section id="diversit√©-de-la-communaut√©" class="level3 page-columns page-full" data-number="26.5.1"><h3 data-number="26.5.1" class="anchored" data-anchor-id="diversit√©-de-la-communaut√©">
<span class="header-section-number">26.5.1</span> Diversit√© de la communaut√©</h3>
<p>La diversit√© de la communaut√© en microbiologie est mesur√©e par plusieurs indices : - la richesse en esp√®ces (nombre total d‚Äôesp√®ces) - l‚Äô√©quitabilit√© (r√©partition des esp√®ces au sein d‚Äôun microbiome) - la diversit√© (combinaison des deux)</p>
<div class="page-columns page-full"><p>Le <strong>coefficient de <span class="citation" data-cites="Hill">Hill (<a href="#ref-Hill" role="doc-biblioref">1910</a>)</span></strong> combine ces mesures en une seule √©quation. Toutes ces variations sont appel√©es diversit√© alpha.</p><div class="no-row-height column-margin column-container"><div id="ref-Hill" class="csl-entry" role="listitem">
Hill, Archibald V. 1910. <span>‚ÄúThe Possible Effects of the Aggregation of the Molecules of H√¶moglobin on Its Dissociation Curves.‚Äù</span> <em>Journal of Physiology</em> 40: iv‚Äìvii. <a href="https://en.wikipedia.org/wiki/Hill_equation_(biochemistry)">https://en.wikipedia.org/wiki/Hill_equation_(biochemistry)</a>.
</div></div></div>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Estimate (observed) richness</span></span>
<span><span class="va">tse_alpha</span> <span class="op">&lt;-</span> <span class="fu">addAlpha</span><span class="op">(</span></span>
<span>    <span class="va">tse</span>,</span>
<span>    assay.type <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>    index <span class="op">=</span> <span class="st">"observed"</span>,</span>
<span>    name<span class="op">=</span><span class="st">"observed"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Check some of the first values in colData</span></span>
<span><span class="va">tse_alpha</span><span class="op">$</span><span class="va">observed</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">##  [1] 68 51 68 62 58 61</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Le r√©sultat montre les valeurs de richesse estim√©es pour diff√©rents √©chantillons ou emplacements au sein de l‚Äôensemble de donn√©es. Il donne une id√©e de la diversit√© de chaque √©chantillon en termes de nombre d‚Äôesp√®ces diff√©rentes pr√©sentes. Nous pouvons ensuite cr√©er un graphique pour visualiser cela.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">plotColData</span><span class="op">(</span></span>
<span>    <span class="va">tse_alpha</span>,</span>
<span>    <span class="st">"observed"</span>,</span>
<span>    <span class="st">"cohort"</span>,</span>
<span>    colour_by <span class="op">=</span> <span class="st">"patient_status"</span><span class="op">)</span> <span class="op">+</span></span>
<span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle<span class="op">=</span><span class="fl">45</span>,hjust<span class="op">=</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>y<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/expression.html">expression</a></span><span class="op">(</span><span class="va">Richness</span><span class="op">[</span><span class="va">Observed</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="introductory_workflow_french_version_files/figure-html/plotColdata-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Pour aller encore plus loin, nous pouvons √©galement comparer l‚Äôindice de Shannon estim√© √† la richesse observ√©e. Shannon quantifie la diversit√© en termes √† la fois du nombre d‚Äôesp√®ces diff√©rentes (richesse) et de l‚Äôuniformit√© de leur r√©partition (abondance) et est calcul√© comme suit :</p>
<p><span class="math display">\[
H' = -\sum_{i=1}^{R} p_i \ln(p_i)
\]</span> p<sub>i</sub> √©tant la proportion d‚Äôun certain microorganisme.</p>
<hr>
<p>D‚Äôabord, nous pouvons facilement calculer cette mesure et l‚Äôajouter √† notre TreeSE.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tse_alpha</span> <span class="op">&lt;-</span> <span class="fu">addAlpha</span><span class="op">(</span></span>
<span>    <span class="va">tse_alpha</span>,</span>
<span>    assay.type <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>    index <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"shannon"</span><span class="op">)</span>,</span>
<span>    name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"shannon"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Nous pouvons √©galement comparer les deux mesures de diversit√© en produisant les graphiques suivants.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Create the plots</span></span>
<span><span class="va">plots</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"observed"</span>, <span class="st">"shannon"</span><span class="op">)</span>,</span>
<span>    <span class="va">plotColData</span>,</span>
<span>    object <span class="op">=</span> <span class="va">tse_alpha</span>,</span>
<span>    x <span class="op">=</span> <span class="st">"patient_status"</span>,</span>
<span>    colour_by <span class="op">=</span> <span class="st">"patient_status"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Fine-tune visual appearance</span></span>
<span><span class="va">plots</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">plots</span>, <span class="st">"+"</span>,</span>
<span>    <span class="fu">theme</span><span class="op">(</span></span>
<span>        axis.text.x <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>        axis.title.x <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>        axis.ticks.x <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot the figures</span></span>
<span><span class="op">(</span><span class="va">plots</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">|</span> <span class="va">plots</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">plot_layout</span><span class="op">(</span>guides <span class="op">=</span> <span class="st">"collect"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="introductory_workflow_french_version_files/figure-html/violinPlots-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Il est tr√®s important de faire toutes ces comparaisons afin de quantifier la diversit√© et de comparer les √©chantillons dans nos donn√©es en utilisant diff√©rentes mesures. - Vous pouvez trouver d‚Äôautres types de comparaisons directement dans le livre <a href="alpha_diversity.html" class="quarto-xref"><span>Chapter 13</span></a>.</p>
<hr></section><section id="similarit√©-de-la-communaut√©" class="level3 page-columns page-full" data-number="26.5.2"><h3 data-number="26.5.2" class="anchored" data-anchor-id="similarit√©-de-la-communaut√©">
<span class="header-section-number">26.5.2</span> Similarit√© de la communaut√©</h3>
<p>La similarit√© de la communaut√© fait r√©f√©rence √† la mani√®re dont les microorganismes se ressemblent en termes de composition et d‚Äôabondance des diff√©rents taxons microbiens. Cela peut nous aider √† comprendre dans quelle mesure diff√©rents √©chantillons se ressemblent et √† trouver des informations cl√©s. Cependant, en analyse de microbiome, il est plus courant de mesurer la dissimilarit√©/diversit√© b√™ta entre deux √©chantillons A et B en utilisant la mesure de Bray-Curtis qui est d√©finie comme suit :</p>
<p><span class="math display">\[
BC_{ij} = \frac{\sum_{k} |A_{k} - B_{k}|}{\sum_{k} (A_{k} + B_{k})}
\]</span></p>
<p>Heureusement pour nous, le <a href="https://microbiome.github.io/mia/"><strong>package mia</strong></a> fournit un moyen facile de calculer l‚Äôabondance relative pour notre TreeSE en utilisant la m√©thode transformAssay.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu">transformAssay</span><span class="op">(</span></span>
<span>    <span class="va">tse</span>,</span>
<span>    assay.type <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>    method <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Cela prendra l‚Äôassay des comptes d‚Äôorigine et appliquera le calcul des abondances relatives. Le r√©sultat est une matrice avec les identifiants des √©chantillons en lignes et les abondances relatives pour chaque taxon dans ces √©chantillons en colonnes. Il peut √™tre consult√© dans les assays du tse :</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">assay</span><span class="op">(</span><span class="va">tse</span>, <span class="st">"relabundance"</span><span class="op">)</span><span class="op">[</span><span class="fl">5</span><span class="op">:</span><span class="fl">10</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span></span>
<span><span class="co">##                       A110      A12     A15      A19      A21    A23      A25</span></span>
<span><span class="co">##  Akkermansia       0.03057 0.046595 0.07539 0.014894 0.013226 0.1282 0.040124</span></span>
<span><span class="co">##  Bacteroides_3     0.00000 0.160112 0.00000 0.113619 0.096048 0.0000 0.047602</span></span>
<span><span class="co">##  Parabacteroides_1 0.00000 0.005470 0.00000 0.003290 0.004409 0.0000 0.002280</span></span>
<span><span class="co">##  Bacteroides_4     0.00000 0.089370 0.00000 0.019663 0.004566 0.0000 0.064654</span></span>
<span><span class="co">##  Bacteroides_5     0.00000 0.066061 0.00000 0.013542 0.006928 0.0000 0.055079</span></span>
<span><span class="co">##  Parabacteroides_2 0.00000 0.004632 0.00000 0.002907 0.005196 0.0000 0.002189</span></span>
<span><span class="co">##                         A28      A29     A34</span></span>
<span><span class="co">##  Akkermansia       0.035930 0.014108 0.07693</span></span>
<span><span class="co">##  Bacteroides_3     0.075247 0.198646 0.00000</span></span>
<span><span class="co">##  Parabacteroides_1 0.003240 0.004515 0.00000</span></span>
<span><span class="co">##  Bacteroides_4     0.033574 0.134312 0.00000</span></span>
<span><span class="co">##  Bacteroides_5     0.064792 0.132054 0.00000</span></span>
<span><span class="co">##  Parabacteroides_2 0.002798 0.003386 0.00000</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Ensuite, nous pouvons ajouter la dissimilarit√© de Bray-Curtis :</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Run PCoA on relabundance assay with Bray-Curtis distances</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu">runMDS</span><span class="op">(</span></span>
<span>    <span class="va">tse</span>,</span>
<span>    FUN <span class="op">=</span> <span class="va">vegdist</span>,</span>
<span>    method <span class="op">=</span> <span class="st">"bray"</span>,</span>
<span>    assay.type <span class="op">=</span> <span class="st">"relabundance"</span>,</span>
<span>    name <span class="op">=</span> <span class="st">"MDS_bray"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Dans notre cas, l‚Äôassay contient 151 lignes et 27 colonnes. Avoir autant de colonnes et donc de dimensions peut √™tre probl√©matique pour visualiser la dissimilarit√©.</p>
<p>Pour visualiser la dissimilarit√© entre les diff√©rents √©chantillons, nous pouvons effectuer une analyse en coordonn√©es principales sur l‚Äôassay nouvellement cr√©√©. Cela projette essentiellement les dimensions de Bray-Curtis sur un espace inf√©rieur tout en conservant autant de variation que possible, les valeurs projet√©es √©tant appel√©es coordonn√©es principales. Vous pouvez en lire plus sur <span class="citation" data-cites="Multidimensional-scaling"><span>‚ÄúMultidimensional Scaling‚Äù</span> (<a href="#ref-Multidimensional-scaling" role="doc-biblioref">n.d.</a>)</span> ici.</p>
<div class="no-row-height column-margin column-container"><div id="ref-Multidimensional-scaling" class="csl-entry" role="listitem">
<span>‚ÄúMultidimensional Scaling.‚Äù</span> n.d. <a href="https://en.wikipedia.org/wiki/Multidimensional_scaling#Types" class="uri">https://en.wikipedia.org/wiki/Multidimensional_scaling#Types</a>.
</div><div id="ref-R_vegan" class="csl-entry" role="listitem">
Oksanen, Jari, F. Guillaume Blanchet, Michael Friendly, Roeland Kindt, Pierre Legendre, Dan McGlinn, Peter R. Minchin, et al. 2020. <em>Vegan: Community Ecology Package</em>. <a href="https://CRAN.R-project.org/package=vegan">https://CRAN.R-project.org/package=vegan</a>.
</div></div><p><code>mia</code> fournit certaines techniques de r√©duction de dimension, telles que dbRDA. De plus, nous pouvons utiliser le package <code>scater</code> de Bioconductor et le package <code>vegan</code>, cr√©√© par <span class="citation" data-cites="R_vegan">Oksanen et al. (<a href="#ref-R_vegan" role="doc-biblioref">2020</a>)</span> pour transformer la dissimilarit√© en distances r√©elles pouvant √™tre visualis√©es :</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Create ggplot object</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">plotReducedDim</span><span class="op">(</span><span class="va">tse</span>, <span class="st">"MDS_bray"</span>,colour_by <span class="op">=</span> <span class="st">"cohort"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Convert to an interactive plot with ggplotly</span></span>
<span><span class="fu">ggplotly</span><span class="op">(</span><span class="va">p</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="plotly html-widget html-fill-item" id="htmlwidget-91b6a2178b523dfb82a5" style="width:100%;height:464px;"></div>
<script type="application/json" data-for="htmlwidget-91b6a2178b523dfb82a5">{"x":{"data":[{"x":[0.35648019510342693,-0.018374142495210428,0.14308108202418346,0.030310752417021462,0.35077181787238798,-0.49577864638760855,0.087394516343590731,-0.19026611013765943,-0.20258006023821226,0.31092036586310173],"y":[0.11979799941456977,-0.35553853106007549,0.36730630483985605,-0.34670413116862575,0.09347807030913835,0.19442644610454576,0.16051336300969038,0.021136801207098842,-0.00081589685663303467,0.17199125072312924],"text":["X:  0.35648<br />Y:  0.1197980<br />colour_by: Cohort_1","X: -0.01837<br />Y: -0.3555385<br />colour_by: Cohort_1","X:  0.14308<br />Y:  0.3673063<br />colour_by: Cohort_1","X:  0.03031<br />Y: -0.3467041<br />colour_by: Cohort_1","X:  0.35077<br />Y:  0.0934781<br />colour_by: Cohort_1","X: -0.49578<br />Y:  0.1944264<br />colour_by: Cohort_1","X:  0.08739<br />Y:  0.1605134<br />colour_by: Cohort_1","X: -0.19027<br />Y:  0.0211368<br />colour_by: Cohort_1","X: -0.20258<br />Y: -0.0008159<br />colour_by: Cohort_1","X:  0.31092<br />Y:  0.1719913<br />colour_by: Cohort_1"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(114,158,206,1)","opacity":0.59999999999999998,"size":5.6692913385826778,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(114,158,206,1)"}},"hoveron":"points","name":"Cohort_1","legendgroup":"Cohort_1","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.075390634272432269,0.17993858815213842,-0.061801267881730078,-0.078201021670979504,-0.11328181884493595,0.11115537834788053,-0.45676030620236741,0.31378598742281399,-0.43492202659263263,0.17893342847928664],"y":[-0.31823525043539219,0.094766740653734735,-0.35999740035119049,-0.38897158823576405,-0.44727748546615315,0.16105988025022688,0.062412413548315719,0.10025830207293528,0.18363500432988161,0.0030736404258236258],"text":["X:  0.07539<br />Y: -0.3182353<br />colour_by: Cohort_2","X:  0.17994<br />Y:  0.0947667<br />colour_by: Cohort_2","X: -0.06180<br />Y: -0.3599974<br />colour_by: Cohort_2","X: -0.07820<br />Y: -0.3889716<br />colour_by: Cohort_2","X: -0.11328<br />Y: -0.4472775<br />colour_by: Cohort_2","X:  0.11116<br />Y:  0.1610599<br />colour_by: Cohort_2","X: -0.45676<br />Y:  0.0624124<br />colour_by: Cohort_2","X:  0.31379<br />Y:  0.1002583<br />colour_by: Cohort_2","X: -0.43492<br />Y:  0.1836350<br />colour_by: Cohort_2","X:  0.17893<br />Y:  0.0030736<br />colour_by: Cohort_2"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,158,74,1)","opacity":0.59999999999999998,"size":5.6692913385826778,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(255,158,74,1)"}},"hoveron":"points","name":"Cohort_2","legendgroup":"Cohort_2","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.28013630859971222,0.29547319435684533,-0.054804951985339558,-0.54913250259596857,0.20758146320583995,-0.48090546548328372,0.21545460805526706],"y":[0.14833722087768203,0.13805107982469711,-0.03270977343545451,0.24268033887409377,-0.092710188965328399,0.11220370540514778,-0.032168315895949241],"text":["X:  0.28014<br />Y:  0.1483372<br />colour_by: Cohort_3","X:  0.29547<br />Y:  0.1380511<br />colour_by: Cohort_3","X: -0.05480<br />Y: -0.0327098<br />colour_by: Cohort_3","X: -0.54913<br />Y:  0.2426803<br />colour_by: Cohort_3","X:  0.20758<br />Y: -0.0927102<br />colour_by: Cohort_3","X: -0.48091<br />Y:  0.1122037<br />colour_by: Cohort_3","X:  0.21545<br />Y: -0.0321683<br />colour_by: Cohort_3"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(103,191,92,1)","opacity":0.59999999999999998,"size":5.6692913385826778,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(103,191,92,1)"}},"hoveron":"points","name":"Cohort_3","legendgroup":"Cohort_3","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":25.298464092984638,"r":6.6417600664176026,"b":37.288738658601673,"l":51.710846231394186},"font":{"color":"rgba(0,0,0,1)","family":"","size":13.283520132835198},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-0.59441313748093838,0.40176082998839668],"tickmode":"array","ticktext":["-0.4","-0.2","0.0","0.2","0.4"],"tickvals":[-0.40000000000000008,-0.20000000000000007,0,0.19999999999999996,0.39999999999999991],"categoryorder":"array","categoryarray":["-0.4","-0.2","0.0","0.2","0.4"],"nticks":null,"ticks":"outside","tickcolor":"rgba(0,0,0,1)","ticklen":3.3208800332087995,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(0,0,0,1)","family":"","size":11.385874399573032},"tickangle":-0,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176002,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":{"text":"MDS_bray 1","font":{"color":"rgba(0,0,0,1)","family":"","size":13.283520132835198}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-0.48800667498145361,0.40803549435515651],"tickmode":"array","ticktext":["-0.25","0.00","0.25"],"tickvals":[-0.25,-5.5511151231257827e-17,0.24999999999999994],"categoryorder":"array","categoryarray":["-0.25","0.00","0.25"],"nticks":null,"ticks":"outside","tickcolor":"rgba(0,0,0,1)","ticklen":3.3208800332088013,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(0,0,0,1)","family":"","size":11.38587439957303},"tickangle":-0,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176002,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":{"text":"MDS_bray 2","font":{"color":"rgba(0,0,0,1)","family":"","size":13.283520132835198}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":true,"legend":{"bgcolor":null,"bordercolor":null,"borderwidth":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.385874399573032},"title":{"text":"cohort","font":{"color":"rgba(0,0,0,1)","family":"","size":13.283520132835198}}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"95d5d9a139b":{"x":{},"y":{},"colour":{},"type":"scatter"}},"cur_data":"95d5d9a139b","visdat":{"95d5d9a139b":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p>Cependant, les axes ne sont pas tr√®s informatifs et la quantit√© de variance captur√©e par l‚Äôalgorithme n‚Äôest nulle part indiqu√©e. Nous pouvons ajuster le graphique pour montrer plus d‚Äôinformations comme suit :</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Calculate explained variance</span></span>
<span><span class="va">e</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/attr.html">attr</a></span><span class="op">(</span><span class="fu">reducedDim</span><span class="op">(</span><span class="va">tse</span>, <span class="st">"MDS_bray"</span><span class="op">)</span>, <span class="st">"eig"</span><span class="op">)</span></span>
<span><span class="va">rel_eig</span> <span class="op">&lt;-</span> <span class="va">e</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">e</span><span class="op">[</span><span class="va">e</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add explained variance for each axis on the plot</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="va">p</span> <span class="op">+</span> <span class="fu">labs</span><span class="op">(</span></span>
<span>    x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"PCoA 1 ("</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fl">100</span> <span class="op">*</span> <span class="va">rel_eig</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="fl">1</span><span class="op">)</span>, <span class="st">"%"</span>, <span class="st">")"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>,</span>
<span>    y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"PCoA 2 ("</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fl">100</span> <span class="op">*</span> <span class="va">rel_eig</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span>, <span class="fl">1</span><span class="op">)</span>, <span class="st">"%"</span>, <span class="st">")"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Reonvert to an interactive plot with ggplotly</span></span>
<span><span class="fu">ggplotly</span><span class="op">(</span><span class="va">p</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="plotly html-widget html-fill-item" id="htmlwidget-ba1123d9f75dc487b90a" style="width:100%;height:464px;"></div>
<script type="application/json" data-for="htmlwidget-ba1123d9f75dc487b90a">{"x":{"data":[{"x":[0.35648019510342693,-0.018374142495210428,0.14308108202418346,0.030310752417021462,0.35077181787238798,-0.49577864638760855,0.087394516343590731,-0.19026611013765943,-0.20258006023821226,0.31092036586310173],"y":[0.11979799941456977,-0.35553853106007549,0.36730630483985605,-0.34670413116862575,0.09347807030913835,0.19442644610454576,0.16051336300969038,0.021136801207098842,-0.00081589685663303467,0.17199125072312924],"text":["X:  0.35648<br />Y:  0.1197980<br />colour_by: Cohort_1","X: -0.01837<br />Y: -0.3555385<br />colour_by: Cohort_1","X:  0.14308<br />Y:  0.3673063<br />colour_by: Cohort_1","X:  0.03031<br />Y: -0.3467041<br />colour_by: Cohort_1","X:  0.35077<br />Y:  0.0934781<br />colour_by: Cohort_1","X: -0.49578<br />Y:  0.1944264<br />colour_by: Cohort_1","X:  0.08739<br />Y:  0.1605134<br />colour_by: Cohort_1","X: -0.19027<br />Y:  0.0211368<br />colour_by: Cohort_1","X: -0.20258<br />Y: -0.0008159<br />colour_by: Cohort_1","X:  0.31092<br />Y:  0.1719913<br />colour_by: Cohort_1"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(114,158,206,1)","opacity":0.59999999999999998,"size":5.6692913385826778,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(114,158,206,1)"}},"hoveron":"points","name":"Cohort_1","legendgroup":"Cohort_1","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.075390634272432269,0.17993858815213842,-0.061801267881730078,-0.078201021670979504,-0.11328181884493595,0.11115537834788053,-0.45676030620236741,0.31378598742281399,-0.43492202659263263,0.17893342847928664],"y":[-0.31823525043539219,0.094766740653734735,-0.35999740035119049,-0.38897158823576405,-0.44727748546615315,0.16105988025022688,0.062412413548315719,0.10025830207293528,0.18363500432988161,0.0030736404258236258],"text":["X:  0.07539<br />Y: -0.3182353<br />colour_by: Cohort_2","X:  0.17994<br />Y:  0.0947667<br />colour_by: Cohort_2","X: -0.06180<br />Y: -0.3599974<br />colour_by: Cohort_2","X: -0.07820<br />Y: -0.3889716<br />colour_by: Cohort_2","X: -0.11328<br />Y: -0.4472775<br />colour_by: Cohort_2","X:  0.11116<br />Y:  0.1610599<br />colour_by: Cohort_2","X: -0.45676<br />Y:  0.0624124<br />colour_by: Cohort_2","X:  0.31379<br />Y:  0.1002583<br />colour_by: Cohort_2","X: -0.43492<br />Y:  0.1836350<br />colour_by: Cohort_2","X:  0.17893<br />Y:  0.0030736<br />colour_by: Cohort_2"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,158,74,1)","opacity":0.59999999999999998,"size":5.6692913385826778,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(255,158,74,1)"}},"hoveron":"points","name":"Cohort_2","legendgroup":"Cohort_2","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.28013630859971222,0.29547319435684533,-0.054804951985339558,-0.54913250259596857,0.20758146320583995,-0.48090546548328372,0.21545460805526706],"y":[0.14833722087768203,0.13805107982469711,-0.03270977343545451,0.24268033887409377,-0.092710188965328399,0.11220370540514778,-0.032168315895949241],"text":["X:  0.28014<br />Y:  0.1483372<br />colour_by: Cohort_3","X:  0.29547<br />Y:  0.1380511<br />colour_by: Cohort_3","X: -0.05480<br />Y: -0.0327098<br />colour_by: Cohort_3","X: -0.54913<br />Y:  0.2426803<br />colour_by: Cohort_3","X:  0.20758<br />Y: -0.0927102<br />colour_by: Cohort_3","X: -0.48091<br />Y:  0.1122037<br />colour_by: Cohort_3","X:  0.21545<br />Y: -0.0321683<br />colour_by: Cohort_3"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(103,191,92,1)","opacity":0.59999999999999998,"size":5.6692913385826778,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(103,191,92,1)"}},"hoveron":"points","name":"Cohort_3","legendgroup":"Cohort_3","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":25.298464092984638,"r":6.6417600664176026,"b":37.288738658601673,"l":51.710846231394186},"font":{"color":"rgba(0,0,0,1)","family":"","size":13.283520132835198},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-0.59441313748093838,0.40176082998839668],"tickmode":"array","ticktext":["-0.4","-0.2","0.0","0.2","0.4"],"tickvals":[-0.40000000000000008,-0.20000000000000007,0,0.19999999999999996,0.39999999999999991],"categoryorder":"array","categoryarray":["-0.4","-0.2","0.0","0.2","0.4"],"nticks":null,"ticks":"outside","tickcolor":"rgba(0,0,0,1)","ticklen":3.3208800332087995,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(0,0,0,1)","family":"","size":11.385874399573032},"tickangle":-0,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176002,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":{"text":"PCoA 1 (36.9%)","font":{"color":"rgba(0,0,0,1)","family":"","size":13.283520132835198}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-0.48800667498145361,0.40803549435515651],"tickmode":"array","ticktext":["-0.25","0.00","0.25"],"tickvals":[-0.25,-5.5511151231257827e-17,0.24999999999999994],"categoryorder":"array","categoryarray":["-0.25","0.00","0.25"],"nticks":null,"ticks":"outside","tickcolor":"rgba(0,0,0,1)","ticklen":3.3208800332088013,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(0,0,0,1)","family":"","size":11.38587439957303},"tickangle":-0,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176002,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":{"text":"PCoA 2 (22.7%)","font":{"color":"rgba(0,0,0,1)","family":"","size":13.283520132835198}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":true,"legend":{"bgcolor":null,"bordercolor":null,"borderwidth":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.385874399573032},"title":{"text":"cohort","font":{"color":"rgba(0,0,0,1)","family":"","size":13.283520132835198}}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"95de2534cc":{"x":{},"y":{},"colour":{},"type":"scatter"}},"cur_data":"95de2534cc","visdat":{"95de2534cc":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p>Et voil√† ! Chaque axe montre la quantit√© de variance ou dans notre cas de dissimilarit√© retenue par chaque coordonn√©e principale. Vous pouvez √©galement ajouter d‚Äôautres options pour colorier par une certaine caract√©ristique par exemple. Vous pouvez en savoir plus dans <a href="beta_diversity.html" class="quarto-xref"><span>Chapter 14</span></a>.</p>
<hr></section></section><section id="visualisation-des-donn√©es" class="level2" data-number="26.6"><h2 data-number="26.6" class="anchored" data-anchor-id="visualisation-des-donn√©es">
<span class="header-section-number">26.6</span> Visualisation des donn√©es</h2>
<p>Les cartes de chaleurs sont l‚Äôun des moyens les plus polyvalents de visualiser vos donn√©es. Dans cette section, nous verrons comment cr√©er une heatmap de base pour visualiser les caract√©ristiques les plus r√©pandues en utilisant la <a href="https://bioconductor.org/packages/release/bioc/html/sechm.html"><strong>biblioth√®que sechm</strong></a>. Pour une carte de chaleur plus d√©taill√©e, veuillez vous reporter √† <strong>cette section</strong> <a href="cross_correlation.html#sec-cross-correlation" class="quarto-xref"><span>Section 20.1</span></a>.</p>
<p>Ensuite, nous allons cr√©er un sous-ensemble de TreeSE pour les taxons les plus r√©pandus en utilisant une exp√©rience alternative :</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">altExp</span><span class="op">(</span><span class="va">tse</span>, <span class="st">"prevalence-subset"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">subsetByPrevalent</span><span class="op">(</span><span class="va">tse</span>,prevalence<span class="op">=</span><span class="fl">0.5</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="op">]</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Lors de la cr√©ation de sous-ensembles avec cette fonction, l‚Äôobjet r√©sultant ne contient plus les abondances relatives correctes car ces abondances ont √©t√© calcul√©es √† l‚Äôorigine sur la basees donn√©es compl√®tes. Par cons√©quent, il est essentiel de recalculer les abondances relatives pour notre sous-ensemble :</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">altExp</span><span class="op">(</span><span class="va">tse</span>, <span class="st">"prevalence-subset"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">transformAssay</span><span class="op">(</span></span>
<span>    <span class="fu">altExp</span><span class="op">(</span><span class="va">tse</span>, <span class="st">"prevalence-subset"</span><span class="op">)</span>,</span>
<span>    assay.type <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>    method <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Maintenant que nous avons pr√©par√© les donn√©es, nous pouvons utiliser la biblioth√®que <code>sechm</code> pr√©c√©demment charg√©e pour visualiser la carte de chaleur :</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Sets the colors</span></span>
<span><span class="fu">setSechmOption</span><span class="op">(</span><span class="st">"hmcols"</span>, value<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#F0F0FF"</span>,<span class="st">"#007562"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plots the actual heatmap.</span></span>
<span><span class="fu">sechm</span><span class="op">(</span></span>
<span>    <span class="fu">altExp</span><span class="op">(</span><span class="va">tse</span>, <span class="st">"prevalence-subset"</span><span class="op">)</span>, features <span class="op">=</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="fu">altExp</span><span class="op">(</span><span class="va">tse</span>, <span class="st">"prevalence-subset"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    assayName<span class="op">=</span><span class="st">"relabundance"</span>,show_colnames<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="introductory_workflow_french_version_files/figure-html/heatmap-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Sur la carte de chaleur ci-dessus, il est √©vident que les Parabacteroides sont relativement fr√©quents dans certains √©chantillons, tandis que les Akkermansia sont d√©tect√©s tr√®s rarement.</p>
</section>
</div>



<a onclick="window.scrollTo(0, 0); return false;" role="button" id="quarto-back-to-top"><i class="bi bi-arrow-up"></i> Back to top</a></main>
</div>


 <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = false;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "Óßã";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="../pages/introductory_workflow.html" class="pagination-link" aria-label="Introductory">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">25</span>&nbsp; <span class="chapter-title">Introductory</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../pages/introductory_workflow_dutch_version.html" class="pagination-link" aria-label="Introductory (Dutch)">
        <span class="nav-page-text"><span class="chapter-number">27</span>&nbsp; <span class="chapter-title">Introductory (Dutch)</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><!-- /content --><footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
<p><a href="./">Orchestrating Microbiome Analysis with Bioconductor</a></p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">
<p>This book was built with <a href="https://github.com/js2264/BiocBook/">BiocBook</a> with <span class="emoji" data-emoji="heart">‚ù§Ô∏è</span></p>
</div>
  </div>
</footer><script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","loop":false,"selector":".lightbox","descPosition":"bottom","openEffect":"zoom"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>


</body></html>